<!doctype html>
<html>
<head>
<meta charset='UTF-8'><meta name='viewport' content='width=device-width initial-scale=1'>
<title>3-Identification of RNA Modifications</title><link href='https://fonts.loli.net/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext' rel='stylesheet' type='text/css' /><style type='text/css'>html {overflow-x: initial !important;}:root { --bg-color: #ffffff; --text-color: #333333; --select-text-bg-color: #B5D6FC; --select-text-font-color: auto; --monospace: "Lucida Console",Consolas,"Courier",monospace; }
html { font-size: 14px; background-color: var(--bg-color); color: var(--text-color); font-family: "Helvetica Neue", Helvetica, Arial, sans-serif; -webkit-font-smoothing: antialiased; }
body { margin: 0px; padding: 0px; height: auto; bottom: 0px; top: 0px; left: 0px; right: 0px; font-size: 1rem; line-height: 1.42857143; overflow-x: hidden; background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; tab-size: 4; background-position: inherit inherit; background-repeat: inherit inherit; }
iframe { margin: auto; }
a.url { word-break: break-all; }
a:active, a:hover { outline: 0px; }
.in-text-selection, ::selection { text-shadow: none; background: var(--select-text-bg-color); color: var(--select-text-font-color); }
#write { margin: 0px auto; height: auto; width: inherit; word-break: normal; word-wrap: break-word; position: relative; white-space: normal; overflow-x: visible; padding-top: 40px; }
#write.first-line-indent p { text-indent: 2em; }
#write.first-line-indent li p, #write.first-line-indent p * { text-indent: 0px; }
#write.first-line-indent li { margin-left: 2em; }
.for-image #write { padding-left: 8px; padding-right: 8px; }
body.typora-export { padding-left: 30px; padding-right: 30px; }
.typora-export .footnote-line, .typora-export li, .typora-export p { white-space: pre-wrap; }
@media screen and (max-width: 500px) { 
  body.typora-export { padding-left: 0px; padding-right: 0px; }
  #write { padding-left: 20px; padding-right: 20px; }
  .CodeMirror-sizer { margin-left: 0px !important; }
  .CodeMirror-gutters { display: none !important; }
}
#write li > figure:last-child { margin-bottom: 0.5rem; }
#write ol, #write ul { position: relative; }
img { max-width: 100%; vertical-align: middle; }
button, input, select, textarea { color: inherit; font-family: inherit; font-size: inherit; font-style: inherit; font-variant-caps: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; }
input[type="checkbox"], input[type="radio"] { line-height: normal; padding: 0px; }
*, ::after, ::before { box-sizing: border-box; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p, #write pre { width: inherit; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p { position: relative; }
p { line-height: inherit; }
h1, h2, h3, h4, h5, h6 { break-after: avoid-page; break-inside: avoid; orphans: 4; }
p { orphans: 4; }
h1 { font-size: 2rem; }
h2 { font-size: 1.8rem; }
h3 { font-size: 1.6rem; }
h4 { font-size: 1.4rem; }
h5 { font-size: 1.2rem; }
h6 { font-size: 1rem; }
.md-math-block, .md-rawblock, h1, h2, h3, h4, h5, h6, p { margin-top: 1rem; margin-bottom: 1rem; }
.hidden { display: none; }
.md-blockmeta { color: rgb(204, 204, 204); font-weight: 700; font-style: italic; }
a { cursor: pointer; }
sup.md-footnote { padding: 2px 4px; background-color: rgba(238, 238, 238, 0.701961); color: rgb(85, 85, 85); border-top-left-radius: 4px; border-top-right-radius: 4px; border-bottom-right-radius: 4px; border-bottom-left-radius: 4px; cursor: pointer; }
sup.md-footnote a, sup.md-footnote a:hover { color: inherit; text-transform: inherit; text-decoration: inherit; }
#write input[type="checkbox"] { cursor: pointer; width: inherit; height: inherit; }
figure { overflow-x: auto; margin: 1.2em 0px; max-width: calc(100% + 16px); padding: 0px; }
figure > table { margin: 0px; }
tr { break-inside: avoid; break-after: auto; }
thead { display: table-header-group; }
table { border-collapse: collapse; border-spacing: 0px; width: 100%; overflow: auto; break-inside: auto; text-align: left; }
table.md-table td { min-width: 32px; }
.CodeMirror-gutters { border-right-width: 0px; background-color: inherit; }
.CodeMirror-linenumber { }
.CodeMirror { text-align: left; }
.CodeMirror-placeholder { opacity: 0.3; }
.CodeMirror pre { padding: 0px 4px; }
.CodeMirror-lines { padding: 0px; }
div.hr:focus { cursor: none; }
#write pre { white-space: pre-wrap; }
#write.fences-no-line-wrapping pre { white-space: pre; }
#write pre.ty-contain-cm { white-space: normal; }
.CodeMirror-gutters { margin-right: 4px; }
.md-fences { font-size: 0.9rem; display: block; break-inside: avoid; text-align: left; overflow: visible; white-space: pre; background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; position: relative !important; background-position: inherit inherit; background-repeat: inherit inherit; }
.md-diagram-panel { width: 100%; margin-top: 10px; text-align: center; padding-top: 0px; padding-bottom: 8px; overflow-x: auto; }
#write .md-fences.mock-cm { white-space: pre-wrap; }
.md-fences.md-fences-with-lineno { padding-left: 0px; }
#write.fences-no-line-wrapping .md-fences.mock-cm { white-space: pre; overflow-x: auto; }
.md-fences.mock-cm.md-fences-with-lineno { padding-left: 8px; }
.CodeMirror-line, twitterwidget { break-inside: avoid; }
.footnotes { opacity: 0.8; font-size: 0.9rem; margin-top: 1em; margin-bottom: 1em; }
.footnotes + .footnotes { margin-top: 0px; }
.md-reset { margin: 0px; padding: 0px; border: 0px; outline: 0px; vertical-align: top; text-decoration: none; text-shadow: none; float: none; position: static; width: auto; height: auto; white-space: nowrap; cursor: inherit; line-height: normal; font-weight: 400; text-align: left; box-sizing: content-box; direction: ltr; background-position: 0px 0px; background-repeat: initial initial; }
li div { padding-top: 0px; }
blockquote { margin: 1rem 0px; }
li .mathjax-block, li p { margin: 0.5rem 0px; }
li { margin: 0px; position: relative; }
blockquote > :last-child { margin-bottom: 0px; }
blockquote > :first-child, li > :first-child { margin-top: 0px; }
.footnotes-area { color: rgb(136, 136, 136); margin-top: 0.714rem; padding-bottom: 0.143rem; white-space: normal; }
#write .footnote-line { white-space: pre-wrap; }
@media print { 
  body, html { border: 1px solid transparent; height: 99%; break-after: avoid-page; break-before: avoid-page; font-variant-ligatures: no-common-ligatures; }
  #write { margin-top: 0px; padding-top: 0px; border-color: transparent !important; }
  .typora-export * { -webkit-print-color-adjust: exact; }
  html.blink-to-pdf { font-size: 13px; }
  .typora-export #write { padding-left: 32px; padding-right: 32px; padding-bottom: 0px; break-after: avoid-page; }
  .typora-export #write::after { height: 0px; }
  .is-mac table { break-inside: avoid; }
}
.footnote-line { margin-top: 0.714em; font-size: 0.7em; }
a img, img a { cursor: pointer; }
pre.md-meta-block { font-size: 0.8rem; min-height: 0.8rem; white-space: pre-wrap; background-color: rgb(204, 204, 204); display: block; overflow-x: hidden; background-position: initial initial; background-repeat: initial initial; }
p > .md-image:only-child:not(.md-img-error) img, p > img:only-child { display: block; margin: auto; }
#write.first-line-indent p > .md-image:only-child:not(.md-img-error) img { left: -2em; position: relative; }
p > .md-image:only-child { display: inline-block; width: 100%; }
#write .MathJax_Display { margin: 0.8em 0px 0px; }
.md-math-block { width: 100%; }
.md-math-block:not(:empty)::after { display: none; }
[contenteditable="true"]:active, [contenteditable="true"]:focus, [contenteditable="false"]:active, [contenteditable="false"]:focus { outline: 0px; box-shadow: none; }
.md-task-list-item { position: relative; list-style-type: none; }
.task-list-item.md-task-list-item { padding-left: 0px; }
.md-task-list-item > input { position: absolute; top: 0px; left: 0px; margin-left: -1.2em; margin-top: calc(1em - 10px); border: none; }
.math { font-size: 1rem; }
.md-toc { min-height: 3.58rem; position: relative; font-size: 0.9rem; border-top-left-radius: 10px; border-top-right-radius: 10px; border-bottom-right-radius: 10px; border-bottom-left-radius: 10px; }
.md-toc-content { position: relative; margin-left: 0px; }
.md-toc-content::after, .md-toc::after { display: none; }
.md-toc-item { display: block; color: rgb(65, 131, 196); }
.md-toc-item a { text-decoration: none; }
.md-toc-inner:hover { text-decoration: underline; }
.md-toc-inner { display: inline-block; cursor: pointer; }
.md-toc-h1 .md-toc-inner { margin-left: 0px; font-weight: 700; }
.md-toc-h2 .md-toc-inner { margin-left: 2em; }
.md-toc-h3 .md-toc-inner { margin-left: 4em; }
.md-toc-h4 .md-toc-inner { margin-left: 6em; }
.md-toc-h5 .md-toc-inner { margin-left: 8em; }
.md-toc-h6 .md-toc-inner { margin-left: 10em; }
@media screen and (max-width: 48em) { 
  .md-toc-h3 .md-toc-inner { margin-left: 3.5em; }
  .md-toc-h4 .md-toc-inner { margin-left: 5em; }
  .md-toc-h5 .md-toc-inner { margin-left: 6.5em; }
  .md-toc-h6 .md-toc-inner { margin-left: 8em; }
}
a.md-toc-inner { font-size: inherit; font-style: inherit; font-weight: inherit; line-height: inherit; }
.footnote-line a:not(.reversefootnote) { color: inherit; }
.md-attr { display: none; }
.md-fn-count::after { content: "."; }
code, pre, samp, tt { font-family: var(--monospace); }
kbd { margin: 0px 0.1em; padding: 0.1em 0.6em; font-size: 0.8em; color: rgb(36, 39, 41); background-color: rgb(255, 255, 255); border: 1px solid rgb(173, 179, 185); border-top-left-radius: 3px; border-top-right-radius: 3px; border-bottom-right-radius: 3px; border-bottom-left-radius: 3px; box-shadow: rgba(12, 13, 14, 0.2) 0px 1px 0px, rgb(255, 255, 255) 0px 0px 0px 2px inset; white-space: nowrap; vertical-align: middle; background-position: initial initial; background-repeat: initial initial; }
.md-comment { color: rgb(162, 127, 3); opacity: 0.8; font-family: var(--monospace); }
code { text-align: left; }
a.md-print-anchor { white-space: pre !important; border: none !important; display: inline-block !important; position: absolute !important; width: 1px !important; right: 0px !important; outline: 0px !important; text-shadow: initial !important; background-position: 0px 0px !important; background-repeat: initial initial !important; }
.md-inline-math .MathJax_SVG .noError { display: none !important; }
.html-for-mac .inline-math-svg .MathJax_SVG { vertical-align: 0.2px; }
.md-math-block .MathJax_SVG_Display { text-align: center; margin: 0px; position: relative; text-indent: 0px; max-width: none; max-height: none; min-height: 0px; min-width: 100%; width: auto; overflow-y: hidden; display: block !important; }
.MathJax_SVG_Display, .md-inline-math .MathJax_SVG_Display { width: auto; margin: inherit; display: inline-block !important; }
.MathJax_SVG .MJX-monospace { font-family: var(--monospace); }
.MathJax_SVG .MJX-sans-serif { font-family: sans-serif; }
.MathJax_SVG { display: inline; font-style: normal; font-weight: 400; line-height: normal; zoom: 90%; text-indent: 0px; text-align: left; text-transform: none; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; direction: ltr; max-width: none; max-height: none; min-width: 0px; min-height: 0px; border: 0px; padding: 0px; margin: 0px; }
.MathJax_SVG * { transition: none; }
.MathJax_SVG_Display svg { vertical-align: middle !important; margin-bottom: 0px !important; margin-top: 0px !important; }
.os-windows.monocolor-emoji .md-emoji { font-family: "Segoe UI Symbol", sans-serif; }
.md-diagram-panel > svg { max-width: 100%; }
[lang="flow"] svg, [lang="mermaid"] svg { max-width: 100%; height: auto; }
[lang="mermaid"] .node text { font-size: 1rem; }
table tr th { border-bottom-width: 0px; }
video { max-width: 100%; display: block; margin: 0px auto; }
iframe { max-width: 100%; width: 100%; border: none; }
.highlight td, .highlight tr { border: 0px; }
svg[id^="mermaidChart"] { line-height: 1em; }
mark { background-color: rgb(255, 255, 0); color: rgb(0, 0, 0); background-position: initial initial; background-repeat: initial initial; }
.md-html-inline .md-plain, .md-html-inline strong, mark .md-inline-math, mark strong { color: inherit; }
mark .md-meta { color: rgb(0, 0, 0); opacity: 0.3 !important; }


:root {
    --side-bar-bg-color: #fafafa;
    --control-text-color: #777;
}

@include-when-export url(https://fonts.loli.net/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext);

/* open-sans-regular - latin-ext_latin */
  /* open-sans-italic - latin-ext_latin */
    /* open-sans-700 - latin-ext_latin */
    /* open-sans-700italic - latin-ext_latin */
  html {
    font-size: 16px;
}

body {
    font-family: "Open Sans","Clear Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
    color: rgb(51, 51, 51);
    line-height: 1.6;
}

#write {
    max-width: 860px;
  	margin: 0 auto;
  	padding: 30px;
    padding-bottom: 100px;
}

@media only screen and (min-width: 1400px) {
	#write {
		max-width: 1024px;
	}
}

@media only screen and (min-width: 1800px) {
	#write {
		max-width: 1200px;
	}
}

#write > ul:first-child,
#write > ol:first-child{
    margin-top: 30px;
}

a {
    color: #4183C4;
}
h1,
h2,
h3,
h4,
h5,
h6 {
    position: relative;
    margin-top: 1rem;
    margin-bottom: 1rem;
    font-weight: bold;
    line-height: 1.4;
    cursor: text;
}
h1:hover a.anchor,
h2:hover a.anchor,
h3:hover a.anchor,
h4:hover a.anchor,
h5:hover a.anchor,
h6:hover a.anchor {
    text-decoration: none;
}
h1 tt,
h1 code {
    font-size: inherit;
}
h2 tt,
h2 code {
    font-size: inherit;
}
h3 tt,
h3 code {
    font-size: inherit;
}
h4 tt,
h4 code {
    font-size: inherit;
}
h5 tt,
h5 code {
    font-size: inherit;
}
h6 tt,
h6 code {
    font-size: inherit;
}
h1 {
    padding-bottom: .3em;
    font-size: 2.25em;
    line-height: 1.2;
    border-bottom: 1px solid #eee;
}
h2 {
   padding-bottom: .3em;
    font-size: 1.75em;
    line-height: 1.225;
    border-bottom: 1px solid #eee;
}
h3 {
    font-size: 1.5em;
    line-height: 1.43;
}
h4 {
    font-size: 1.25em;
}
h5 {
    font-size: 1em;
}
h6 {
   font-size: 1em;
    color: #777;
}
p,
blockquote,
ul,
ol,
dl,
table{
    margin: 0.8em 0;
}
li>ol,
li>ul {
    margin: 0 0;
}
hr {
    height: 2px;
    padding: 0;
    margin: 16px 0;
    background-color: #e7e7e7;
    border: 0 none;
    overflow: hidden;
    box-sizing: content-box;
}

li p.first {
    display: inline-block;
}
ul,
ol {
    padding-left: 30px;
}
ul:first-child,
ol:first-child {
    margin-top: 0;
}
ul:last-child,
ol:last-child {
    margin-bottom: 0;
}
blockquote {
    border-left: 4px solid #dfe2e5;
    padding: 0 15px;
    color: #777777;
}
blockquote blockquote {
    padding-right: 0;
}
table {
    padding: 0;
    word-break: initial;
}
table tr {
    border-top: 1px solid #dfe2e5;
    margin: 0;
    padding: 0;
}
table tr:nth-child(2n),
thead {
    background-color: #f8f8f8;
}
table tr th {
    font-weight: bold;
    border: 1px solid #dfe2e5;
    border-bottom: 0;
    margin: 0;
    padding: 6px 13px;
}
table tr td {
    border: 1px solid #dfe2e5;
    margin: 0;
    padding: 6px 13px;
}
table tr th:first-child,
table tr td:first-child {
    margin-top: 0;
}
table tr th:last-child,
table tr td:last-child {
    margin-bottom: 0;
}

.CodeMirror-lines {
    padding-left: 4px;
}

.code-tooltip {
    box-shadow: 0 1px 1px 0 rgba(0,28,36,.3);
    border-top: 1px solid #eef2f2;
}

.md-fences,
code,
tt {
    border: 1px solid #e7eaed;
    background-color: #f8f8f8;
    border-radius: 3px;
    padding: 0;
    padding: 2px 4px 0px 4px;
    font-size: 0.9em;
}

code {
    background-color: #f3f4f4;
    padding: 0 2px 0 2px;
}

.md-fences {
    margin-bottom: 15px;
    margin-top: 15px;
    padding-top: 8px;
    padding-bottom: 6px;
}


.md-task-list-item > input {
  margin-left: -1.3em;
}

@media print {
    html {
        font-size: 13px;
    }
    table,
    pre {
        page-break-inside: avoid;
    }
    pre {
        word-wrap: break-word;
    }
}

.md-fences {
	background-color: #f8f8f8;
}
#write pre.md-meta-block {
	padding: 1rem;
    font-size: 85%;
    line-height: 1.45;
    background-color: #f7f7f7;
    border: 0;
    border-radius: 3px;
    color: #777777;
    margin-top: 0 !important;
}

.mathjax-block>.code-tooltip {
	bottom: .375rem;
}

.md-mathjax-midline {
    background: #fafafa;
}

#write>h3.md-focus:before{
	left: -1.5625rem;
	top: .375rem;
}
#write>h4.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
#write>h5.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
#write>h6.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
.md-image>.md-meta {
    /*border: 1px solid #ddd;*/
    border-radius: 3px;
    padding: 2px 0px 0px 4px;
    font-size: 0.9em;
    color: inherit;
}

.md-tag {
    color: #a7a7a7;
    opacity: 1;
}

.md-toc { 
    margin-top:20px;
    padding-bottom:20px;
}

.sidebar-tabs {
    border-bottom: none;
}

#typora-quick-open {
    border: 1px solid #ddd;
    background-color: #f8f8f8;
}

#typora-quick-open-item {
    background-color: #FAFAFA;
    border-color: #FEFEFE #e5e5e5 #e5e5e5 #eee;
    border-style: solid;
    border-width: 1px;
}

/** focus mode */
.on-focus-mode blockquote {
    border-left-color: rgba(85, 85, 85, 0.12);
}

header, .context-menu, .megamenu-content, footer{
    font-family: "Segoe UI", "Arial", sans-serif;
}

.file-node-content:hover .file-node-icon,
.file-node-content:hover .file-node-open-state{
    visibility: visible;
}

.mac-seamless-mode #typora-sidebar {
    background-color: #fafafa;
    background-color: var(--side-bar-bg-color);
}

.md-lang {
    color: #b4654d;
}

.html-for-mac .context-menu {
    --item-hover-bg-color: #E6F0FE;
}

#md-notification .btn {
    border: 0;
}

.dropdown-menu .divider {
    border-color: #e5e5e5;
}

.ty-preferences .window-content {
    background-color: #fafafa;
}

.ty-preferences .nav-group-item.active {
    color: white;
    background: #999;
}

 .typora-export li, .typora-export p, .typora-export,  .footnote-line {white-space: normal;} 
</style>
</head>
<body class='typora-export' >
<div  id='write'  class = ''><h2><a name="0-introduction-for-identification-of-rna-modifications" class="md-header-anchor"></a><span>0. Introduction for Identification of RNA Modifications</span></h2><p><span>This module provides step-by-step functions required for epitranscriptome reads mapping and identification of RNA modifications.</span></p><h4><a name="align-reads-to-genome" class="md-header-anchor"></a><span>Align Reads to Genome</span></h4><p><span>Several commonly used aligners are wrapped to align epitranscriptome reads to genome. Currently, </span><a href="https://ccb.jhu.edu/software/tophat/index.shtml" target="_blank"><span>Tophat2</span></a><span>, </span><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" target="_blank"><span>Bowtie2</span></a><span>, </span><a href="https://github.com/alexdobin/STAR" target="_blank"><span>STAR</span></a><span>, </span><a href="https://www.ccb.jhu.edu/software/hisat/index.shtml" target="_blank"><span>HISAT2</span></a><span>, </span><a href="https://bio-bwa.sourceforge.net/bwa.shtml" target="_blank"><span>bwa-mem</span></a><span>.</span></p><table border="2" align="center" cellspacing="0" cellpadding="" id="grid4">
					<tbody><tr>
						<td><b>Tools</b></td>
						<td><b>Description</b></td>
						<td><b>Input</b></td>
						<td><b>Output</b></td>
						<td><b>Time (test data)</b></td>
						<td><b>Reference</b></td>
					</tr>
					<tr>
						<td><strong>Tophat2</strong></td>
						<td>Tophat2 is a spliced aligner, which aligns short reads by calling Bowtie2 but alows for variable-length indels with respect to the reference genome.</td>
						<td rowspan="5">Epitranscriptome sequencing reads in FASTQ format and reference genome sequences in FASTA format</td>
						<td rowspan="5">Read alignments in SAM/BAM format</td>
						<td>~50s</td>
						<td><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36" target="_blank">Kim <em>et al</em>., 2013, Genome Biology</a></td>
					</tr>
					<tr>
						<td><strong>Bowtie2</strong></td>
						<td>Bowtie2 is a short read aligner which achieves a combination of high speed, sensitivity and accuracy by combining the strengths of the full-text minute index with the flexibility and speed of hardware-accelerated dynamic programming algorithms, therefore bowtie2 is suitable for large genomes</td>
						<td>~10 s</td>
						<td><a href="https://www.nature.com/articles/nmeth.1923" target="_blank">Langmead <em>et al</em>., 2012, Nature Methods</a></td>
					</tr>
					<tr>
						<td><strong>STAR</strong></td>
						<td>STAR is an ultrafast universal RNA-Seq aligner and can discover non-canonical splices and chimeric (fusion) transcripts</td>
						<td>~16s</td>
						<td><a href="https://academic.oup.com/bioinformatics/article/29/1/15/272537" target="_blank">Dobin <em>et al</em>., 2013, Bioinformatics</a></td>
				   </tr>
				   <tr>
						<td><strong>HISAT2</strong></td>
						<td>HISAT2 is an ultrafast spliced aligner with low memory requirements. It supports genomes of any size, including those larger than 4 billion bases</td>
						<td>~8s</td>
						<td><a href="https://www.nature.com/articles/nmeth.3317" target="_blank">Kim <em>et al</em>., 2015, Nature Methods</a></td>
				   </tr>
				   <tr>
						<td><strong>bwa-mem</strong></td>
						<td>bwa-mem is a relatively early aligner based on backward search with Burrows–Wheeler Transform</td>
						<td>~10s</td>
						<td><a href="https://academic.oup.com/bioinformatics/article/25/14/1754/225615" target="_blank">Li <em>et al</em>., 2009, Bioinformatics</a></td>
				   </tr>
</tbody></table><p>&nbsp;</p><h4><a name="identify-rna-modifications" class="md-header-anchor"></a><span>Identify RNA Modifications</span></h4><p><strong><span>Identify RNA Modifications</span></strong><span> implements three pipelines for MeRIP-Seq, CeU-Seq and RNA-BSSeq, respectively.</span></p><figure><table><thead><tr><th><strong><span>Tools</span></strong></th><th><strong><span>Description</span></strong></th><th><strong><span>Input</span></strong></th><th><strong><span>Output</span></strong></th><th><strong><span>Time (test data)</span></strong></th><th><strong><span>Reference</span></strong></th></tr></thead><tbody><tr><td><strong><span>Peak Calling from the MeRIP-Seq data</span></strong></td><td><span>Identify enriched genomic regions from MeRIP-Seq experiment</span></td><td><span>Read alignments of IP and input in SAM/BAM format and reference genome sequences in FASTA format</span></td><td><span>RNA modifications in BED format</span></td><td><span>~36s</span></td><td><a href="https://academic.oup.com/bioinformatics/article/34/21/3747/5021690" target="_blank"><span>Zhai </span><I><span>et al</span></I><span>., 2018, Bioinformatics</span></a></td></tr><tr><td><strong><span>Calling m</span><sup><span>5</span></sup><span>C from the RNA-BSseq data</span></strong></td><td><span>Perform bisulfite sequencing (BS-Seq) read mapping, comprehensive methylation calling using meRanTK</span></td><td><span>Sequencing reads in FASTQ format and reference genome sequences in FASTA format</span></td><td><span>m</span><sup><span>5</span></sup><span>C sites in BED format</span></td><td><span>~10 mins using 2 threads</span></td><td><a href="https://academic.oup.com/bioinformatics/article/32/5/782/1744216" target="_blank"><span>Rieder </span><I><span>et al</span></I><span>., 2016, Bioinformatics</span></a></td></tr><tr><td><strong><span>Calling Ψ from CeU-Seq data</span></strong></td><td><span>Identify pseudouridylation from CeU-Seq</span></td><td><span>Read alignments in SAM/BAM format and cDNA sequences in FASTA format</span></td><td><span>Pseudoridylation sites in BED format</span></td><td><span>~1 mins</span></td><td><a href="https://www.nature.com/articles/nchembio.1836" target="_blank"><span>Li </span><I><span>et al</span></I><span>., 2015, Nature Chemical Biology</span></a></td></tr></tbody></table></figure><h2><a name="1-align-reads-to-genome" class="md-header-anchor"></a><span>1. Align reads to genome</span></h2><p><span>Currently, deepEA wrapped five aligners to map epitranscriptome reads to genome, here, we take </span><a href="https://ccb.jhu.edu/software/tophat/index.shtml" target="_blank"><span>Tophat2</span></a><span> as an example to show how to use deepEA to run reads mapping, the other four aligners are similar.</span></p><h4><a name="input" class="md-header-anchor"></a><span>Input</span></h4><ul><li><strong><span>Epitranscriptome sequencing reads in FASTQ format</span></strong><span> </span></li><li><strong><span>Reference genome in FASTA format</span></strong><span> </span></li></ul><h4><a name="output" class="md-header-anchor"></a><span>Output</span></h4><ul><li><span>Alignments in BAM format</span></li><li><span>Alignment summary generated by tophat2</span></li></ul><h4><a name="how-to-use-this-function" class="md-header-anchor"></a><span>How to use this function</span></h4><ul><li><p><strong><span>Step 1</span></strong><span>: upload the data in directory </span><code>test_data/Identification_of_RNA_Modifications/Align_Reads_to_Genome/</code><span> to history panel, if you are not clear about how to upload local data to deepEA server, please see </span><a href="https://deepea.nwafu.edu.cn/static/tutorial/2-Quality%20Control.html" target="_blank"><span>here</span></a><span> for details</span></p></li><li><p><strong><span>Step 2</span></strong><span>: see the following screenshot to run this function</span></p><p><img src="../assets/img/3-1.png" referrerpolicy="no-referrer" alt="3-1"></p></li></ul><h2><a name="2-peak-calling-from-the-merip-seq-data" class="md-header-anchor"></a><span>2. Peak calling from the MeRIP-Seq data</span></h2><p><strong><span>Peak calling</span></strong><span> is used to identify enriched genomic regions in MeRIP-seq or ChIP-seq experiments. The function is implemented using the </span><strong><span>peakCalling</span></strong><span> function in PEA package (zhai </span><em><span>et al</span></em><span>., 2018)</span></p><h4><a name="input-n2300" class="md-header-anchor"></a><span>Input</span></h4><ul><li><strong><span>IP sample:</span></strong><span> The IP experiment in BAM format</span></li><li><strong><span>Input sample:</span></strong><span> The input control experiment in BAM format</span></li><li><strong><span>Reference genome:</span></strong><span> The Reference genome sequences with FASTA format</span></li><li><strong><span>Reference annotation file:</span></strong><span> The Reference genome annotation file with GTF/GFF3 format (required for methods: </span><strong><span>exomePeak</span></strong><span>, </span><strong><span>MeTPeak</span></strong><span> and </span><strong><span>BayesPeak</span></strong><span>)</span></li></ul><h4><a name="output-n2310" class="md-header-anchor"></a><span>Output</span></h4><ul><li><p><strong><span>The enriched peak region matrix in BED format</span></strong></p><ul><li><p><span>For </span><strong><span>SlidingWindow</span></strong><span> method:</span></p><figure><table><thead><tr><th><span>Chromosome</span></th><th><span>Start(1-based)</span></th><th><span>End</span></th><th><span>Bin number</span></th><th><span>Mean FDR</span></th><th><span>Max FDR</span></th><th><span>Minimum FDR</span></th><th><span>Mean Ratio</span></th><th><span>Max Ratio</span></th><th><span>Minimum Ratio</span></th></tr></thead><tbody><tr><td><span>1</span></td><td><span>67476</span></td><td><span>67575</span></td><td><span>4</span></td><td><span>0.0136</span></td><td><span>0.0328</span></td><td><span>0.0001</span></td><td><span>-1.0012</span></td><td><span>-0.6334</span></td><td><span>-1.581</span></td></tr><tr><td><span>1</span></td><td><span>330776</span></td><td><span>330875</span></td><td><span>4</span></td><td><span>0.0215</span></td><td><span>0.0381</span></td><td><span>0.0007</span></td><td><span>-1.576</span></td><td><span>-1.4077</span></td><td><span>-1.788</span></td></tr><tr><td><span>1</span></td><td><span>389201</span></td><td><span>389300</span></td><td><span>4</span></td><td><span>0.0024</span></td><td><span>0.0070</span></td><td><span>0.0002</span></td><td><span>-1.115</span></td><td><span>-1.0598</span></td><td><span>-1.190</span></td></tr></tbody></table></figure></li><li><p><span>For </span><strong><span>exomePeak</span></strong><span> metod:</span></p><figure><table><thead><tr><th><span>Chromosome</span></th><th><span>Start (0-based)</span></th><th><span>End</span></th><th><span>Gene ID</span></th><th><span>P.value</span></th><th><span>Strand</span></th></tr></thead><tbody><tr><td><span>1</span></td><td><span>30663</span></td><td><span>30723</span></td><td><span>AT1G01040</span></td><td><span>0.0026</span></td><td><span>+</span></td></tr><tr><td><span>1</span></td><td><span>73831</span></td><td><span>74096</span></td><td><span>AT1G01160</span></td><td><span>2.5e-30</span></td><td><span>+</span></td></tr><tr><td><span>1</span></td><td><span>117530</span></td><td><span>117710</span></td><td><span>AT1G01300</span></td><td><span>2.4e-07</span></td><td><span>+</span></td></tr></tbody></table></figure></li><li><p><span>For </span><strong><span>MetPeak</span></strong><span> method: it&#39;s the same as </span><strong><span>exomePeak</span></strong></p></li><li><p><span>For </span><strong><span>BayesPeak</span></strong><span> method:</span></p><figure><table><thead><tr><th><span>chr</span></th><th><span>start</span></th><th><span>end</span></th><th><span>PP</span></th><th><span>job</span></th></tr></thead><tbody><tr><td><span>1</span></td><td><span>3748</span></td><td><span>3848</span></td><td><span>0.0231</span></td><td><span>2</span></td></tr><tr><td><span>1</span></td><td><span>6848</span></td><td><span>6948</span></td><td><span>0.0178</span></td><td><span>2</span></td></tr><tr><td><span>1</span></td><td><span>6898</span></td><td><span>6998</span></td><td><span>0.9960</span></td><td><span>1</span></td></tr></tbody></table></figure></li><li><p><span>For </span><strong><span>macs2</span></strong><span> method: please see </span><a href="https://github.com/macs3-project/MACS" target="_blank"><span>macs2</span></a></p></li></ul></li></ul><h4><a name="how-to-use-this-function-n2424" class="md-header-anchor"></a><span>How to use this function</span></h4><ul><li><strong><span>Step 1</span></strong><span>: upload the data in directory </span><code>test_data/Identification_of_RNA_Modifications/Peak Calling from the MeRIP-Seq data/</code><span> to history panel, if you are not clear about how to upload local data to deepEA server, please see </span><a href="https://deepea.nwafu.edu.cn/static/tutorial/2-Quality%20Control.html" target="_blank"><span>here</span></a><span> for details</span></li><li><strong><span>Step 2</span></strong><span>: see the following screenshot to run this function</span><img src="../assets/img/3-2.png" referrerpolicy="no-referrer" alt="3-2"></li></ul><h2><a name="3-calling-msup5supc-from-the-rna-bsseq-data" class="md-header-anchor"></a><span>3. Calling m</span><sup><span>5</span></sup><span>C from the RNA-BSseq data</span></h2><p><span>This function integrated meRanTK (Rieder </span><em><span>et al</span></em><span>., 2016, </span><em><span>Bioinformatics</span></em><span>) to perform RNA bisulfite sequencing (BS-Seq) read mapping, comprehensive methylation calling.</span></p><h4><a name="input-n2432" class="md-header-anchor"></a><span>Input</span></h4><ul><li><strong><span>FASTQ file:</span></strong><span> The FASTQ format sequencing file</span></li></ul><h4><a name="output-n2436" class="md-header-anchor"></a><span>Output</span></h4><ul><li><strong><span>m5C_out_peaks:</span></strong><span> The detected m</span><sup><span>5</span></sup><span>C sites</span></li></ul><h4><a name="how-to-use-this-function-n2440" class="md-header-anchor"></a><span>How to use this function</span></h4><ul><li><strong><span>Step 1</span></strong><span>: upload the data in directory </span><code>test_data/Identification_of_RNA_Modifications/Calling m5C from the RNA-BSseq data/</code><span> to history panel, if you are not clear about how to upload local data to deepEA server, please see </span><a href="https://deepea.nwafu.edu.cn/static/tutorial/2-Quality%20Control.html" target="_blank"><span>here</span></a><span> for details</span></li><li><strong><span>Step 2</span></strong><span>: see the following screenshot to run this function</span>
<img src="../assets/img/3-3.png" referrerpolicy="no-referrer" alt="3-3"></li></ul><h2><a name="4-calling-ψ-from-ceu-seq-data" class="md-header-anchor"></a><span>4. Calling Ψ from CeU-Seq data</span></h2><p><span>This function is used to identify pseudouridylation from CeU-Seq (Li </span><em><span>et al</span></em><span>., 2015). To be specific, for any given position on a reference transcript, the stop rate of position i was calculated using the equation N_i_stop/(N_i_stop + N_i_readthrough), where N_i_stop (stop reads) is the number of reads with the mapping position starting at base i+1 (one nucleotide 3′ to position i), and N_i_readthrough (readthrough reads) is the number of reads reading through position i; Then a position i is identified to be Ψ only when all of the following criteria were met:</span></p><ul><li><span>the stop reads of position i (N_i_stop) must be no less than 5 in the N3-CMC(+) sample;</span></li><li><span>the stop rate in N3-CMC(−) samples must be less than 0.10;</span></li><li><span>the difference of stop rate for position i between the N3-CMC(+) samples and the matched N3-CMC(−) samples must be at least 0.30.</span></li></ul><h4><a name="input-n2455" class="md-header-anchor"></a><span>Input</span></h4><ul><li><strong><span>Pulldown sample in BAM format:</span></strong><span> The pulldown sample in BAM format</span></li><li><strong><span>Input sample in BAM format:</span></strong><span> The input sample in BAM format</span></li><li><strong><span>Input transcriptome in FASTA format:</span></strong><span> The transcriptome in FASTA format</span></li></ul><h4><a name="output-n2463" class="md-header-anchor"></a><span>Output</span></h4><ul><li><span>A matrix containing the candidate pseudouridine sites</span></li></ul><h4><a name="how-to-use-this-function-n2467" class="md-header-anchor"></a><span>How to use this function</span></h4><ul><li><strong><span>Step 1</span></strong><span>: upload the data in directory </span><code>test_data/Identification_of_RNA_Modifications/Calling pseudouridylation from CeU-Seq/</code><span> to history panel, if you are not clear about how to upload local data to deepEA server, please see </span><a href='2-Quality Control.md'><span>here</span></a><span> for details</span></li><li><strong><span>Step 2</span></strong><span>: see the following screenshot to run this function</span><img src="../assets/img/3-4.png" referrerpolicy="no-referrer" alt="3-4"></li></ul></div>
</body>
</html>