---
layout: default
---

# About
![docker pulls](https://img.shields.io/docker/pulls/malab/pea.svg)
![docker stars](https://img.shields.io/docker/stars/malab/pea.svg)

DeepEA is a Galaxy-based framework for exploring large-scale analysis of epitranscriptome sequencing data. DeepEA is an upgrated version of PEA, which is a special designed R toolkit for plant epitranscriptome analysis. In the current version of DeepEA, a variety of modules and functions including Data Preparation, Quality Control, CMR (chemical modifications of RNA) profiling and analysis, CMR annotation, CMR prediction and related visualization are implemented to enable users perform large-scale, reproducible epitranscriptome analysis. In addition, by taking advantages of multiple instance learning (MIL) algorithms, DeepEA can precisely localize CMRs in peak regions. This can reduce the false positive CMRs generated by MeRIP-Seq/m6A-Seq/m5C-RIP-Seq, as which can localize CMRs to transcript regions of 100-200 nucleotide long but fail to accurately identify CMRs at single-nucleotide solution. The deePEA project is hosted on https://github.com/cma2015/DeepEA. The DeepEA web server can accessed via http://bioinfo.nwafu.edu.cn:4006.

<div id="top"></div>

<!-- [<img align="left" src="assets/img/deNBI_logo.jpg" height="35px" alt="de.NBI" valign="middle">](http://www.denbi.de)
[<img align="center" src="assets/img/elixir_germany.png" height="55px" alt="ELIXIR Germany" valign="middle">](https://www.elixir-europe.org) -->

- [Installation](#installation)
	- [Docker installation](#docker-installation)
	- [DeepEA quick start](#deepea-quick-start)
- [Pre-analysis](pre-analysis)
	- [Data Preparation](#data-preparation)
	- [Quality control](#quality-control)
- [Core-analysis](#core-analysis)
	- [CMR Profiling](#cmr-profiling)
		- [Align Reads to Genome](#align-reads-to-genome)
		- [CMR Calling](#cmr-calling)
		- [Detect Differential CMR Regions](#detect-differential-cmr-regions)
    - [Machine Learning-based CMR Prediction](#machine-learning-based-cmr-prediction)
        - [Feature encoding](#feature-encoding)
        - [Feature selection](#feature-selection)
        - [Machine learning-based classification](#machine-learning-based-classification)
- [Advanced-analysis](#advanced-analysis)
	- [CMR Annotation and Visualization](#cmr-annotation-and-visualization)
    - [Multi-omics Integration Analysis](#multi-omics-integration-analysis)
- [Examples](#examples)
- [How to access help](#how-to-access-help)

# Installation
## Docker installation

> Official installation tutorial is available at [here](https://docs.docker.com/install)

### For Windows (Test on Windows 10 Enterprise version):
* Download [Docker](<https://download.docker.com/win/stable/Docker%20for%20Windows%20Installer.exe>) for windows </br>
* Double click the EXE file to open it;
* Follow the wizard instruction and complete installation;
* Search docker, select ___Docker for Windows___ in the search results and clickit.

### For Mac OS X (Test on macOS Sierra version 10.12.6 and macOS High Sierra version 10.13.3):
* Download [Docker](<https://download.docker.com/mac/stable/Docker.dmg>) for Mac OS X
* Double click the DMG file to open it;
* Drag the docker into Applications and complete installation;
* Start docker from Launchpad by click it.

### For Ubuntu (Test on Ubuntu 16.04 LTS):

```bash
# Update the apt package index:
sudo apt-get update
# Install packages to allow apt to use a repository over HTTPS:
sudo apt-get install \
    apt-transport-https \
    ca-certificates \
    curl \
    software-properties-common
# Add Docker’s official GPG key
curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
sudo apt-key fingerprint 0EBFCD88
sudo add-apt-repository \
   "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
   $(lsb_release -cs) \
   stable"
sudo apt-get update
sudo apt-get install docker-
# Verify that Docker CE is installed correctly by running the hello-world image
sudo docker run hello-world
```

### Verify if Docker is installed correctly

Once Docker installation is completed, we can run ____hello-world____ image to verify if Docker is installed correctly. Open terminal in Mac OS X and Linux operating system and open CMD for Windows operating system, then type the following command:

```bash
$ docker run hello-world
```
**<font color =red>Note</font>:** root permission is required for Linux operating system, if you want to manage docker as a non-root user, please refer [here](https://docs.docker.com/install/linux/linux-postinstall)



## DeepEA quick start

DeepEA has been packaged into a docker image, which can be easily installed as a local server by following command.

```bash
# pull DeepEA from docker hub
docker pull malab/deepea
docker run -it malab/deepea /bin/bash
bash run.sh
```

Then DeepEA local server can be accessed via `http://localhost:8080`

# Pre-analysis

**Pre-analysis** module is composed of two sub-modules: [Data Preparation](#data-preparation) and [Quality Control](#quality-control). These two sub-modules make up the important components of Epitranscriptome.

## Data Preparation

This sub-module provides three funcitons (see following table for details) to prepare epitranscriptome data.
<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Tools</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>References</b></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Obtain_Genome_or_Transcriptome_Sequences_and_Annotation&version=18.09&__identifer=dclshuq6eet">Obtain Genome Sequences and Annotation</a></td>
        <td>Fetch genome, cDNA, CDS sequences and annotation from Enseml automatically</td>
        <td>Select an Ensembl database; Select a species; Select dada type</td>
        <td>FASTA format sequences or GTF annotation</td>
        <td>In-house scripts</td>
   </tr>
   <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Obtain_Epitranscriptome_Sequencing_Reads&version=18.09&__identifer=3bj51oj9isq">Obtain Epitranscriptome Sequencing and Reads</a></td>
        <td>Fetch epitranscriptome sequencing reads from NCBI's SRA database</td>
        <td>SRR accession or HTTP/FTP link</td>
        <td>Sequencing files in SRA/FASTQ format</td>
        <td><a href="https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft">SRA Toolkit</a></td>
   </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Convert_Format&version=18.09&__identifer=uib940h4pb">Sequence Data Preprocessing</a></td>
        <td>xxx</td>
        <td>xxx</td>
        <td>xxx</td>
        <td>xxx</td>
   </tr>
 </table>
</div>

## Quality Control
**Quality Control** submodule consisted of a suite of tools focused on different levels of quality assessment including reads quality, alignment quality and CMR quality, Thus, three functions are implemented including **Assess Reads Quality**, **Assess Alignment Quality** and **Assess CMR quality**. The details of as follows:

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Tool</b></td>
        <td><b>Description</b></td>
        <td><b>Main functions</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>Programs</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td><a href='http://bioinfo.nwafu.edu.cn:4006/?tool_id=Assess_Reads_Quality&amp;version=18.09&amp;__identifer=gr96n8z0sag'>Assess Reads Quality</a></td>
        <td>This tool integrates FastQC and fastp to perform quality control from raw epitranscriptome sequencing reads</td>
        <td>Reads trimming and generate reads quality report including sequence quality, GC content, etc</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Clean reads or reads quality report in HTML format</td>
        <td><a href='https://www.bioinformatics.babraham.ac.uk/projects/fastqc/'>FastQC</a>,<a href='https://github.com/OpenGene/fastp'>fastp</a></td>
        <td><a href='https://doi.org/10.1093/bioinformatics/bty560'>Chen <em>et al</em>., 2018, Bioinformatics</a>, <a href='http://www.bioinformatics.babraham.ac.uk/projects/fastqc'>Babraham Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href='http://bioinfo.nwafu.edu.cn:4006/?tool_id=Assess_Alignment_Quality&amp;version=18.01&amp;__identifer=3uvvorrbx2s'>Assess Alignment Quality</a></td>
        <td>This function is used to generate quality assessment.</td>
        <td>Evaluate alignment quality</td>
        <td>Reads alignment fies in SAM/BAM format</td>
        <td>Alignment quality report in HTML format</td>
        <td><a href='https://github.com/skyhorsetomoon/Trumpet'>trumpet</a></td>
        <td><a href='https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2266-3'>Zhang et al., 2018, BMC Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href='http://bioinfo.nwafu.edu.cn:4006/?tool_id=Assess_CMR_Regions_Quality&amp;version=18.01&amp;__identifer=yu0s7b4tvw7'>Assess CMR Quality</a></td>
        <td>This tool aims to quantify CMR signal strength by counting reads and calculating RPKM in binding site intervals</td>
        <td>Quantify CMRs</td>
        <td>Alignments in SAM/BAM format</td>
        <td>CMR quantification matrix</td>
        <td><a href='https://bioconductor.org/packages/release/bioc/html/DiffBind.html'>DiffBind</a></td>
        <td><a href='https://www.frontiersin.org/articles/10.3389/fgene.2015.00169/full'>Wu <em>et al</em>., 2016, frontiers in Genetics</a></td>
    </tr>
</table>
</div>

# Core-analysis
**Core-analysis** module consists of **CMR Profiling** and **Machine Learning-based CMR Prediction**, of which **CMR Profiling** provides step-by-step functions required for epitranscriptome reads mapping, CMR Calling and differential CMR analysis, while **Machine Learning-based CMR Prediction** is designed to automatically complete the three main steps for constructing a CMR predictor including **Sequence Feature Extraction**, **Feature Engineering** and **Machine Learning-based Classification**

## Align Reads to Genome
Several commonly used aligners are wrapped to align epitranscriptome reads to genome. Currently, [Tophat2](https://ccb.jhu.edu/software/tophat/index.shtml), [Bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml), [STAR](https://github.com/alexdobin/STAR), [HISAT2](http://www.ccb.jhu.edu/software/hisat/index.shtml) and [bwa-mem](http://bio-bwa.sourceforge.net/bwa.shtml) are available.

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Tools</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>References</b></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=tophat2&version=18.09&__identifer=jztsfk48b9">Tophat2</a></td>
        <td>Accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Alignments in SAM/BAM format</td>
        <td><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36">Kim <em>et al</em>., 2013, Genome Biology</a></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=bowtie2&version=18.09&__identifer=x9j53etnvzb">Bowtie2</a></td>
        <td>A fast gapped-read alignment tool</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Alignments in SAM/BAM format</td>
        <td><a href="https://www.nature.com/articles/nmeth.1923">Langmead <em>et al</em>., 2012, Nature Methods</a></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=star&version=18.09&__identifer=gd64qwrc1pp">STAR</a></td>
        <td>An ultrafast universal RNA-seq aligner</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Alignments in SAM/BAM format</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/29/1/15/272537">Dobin <em>et al</em>., 2013, Bioinformatics</a></td>
   </tr>
   <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=hisat2&version=18.09&__identifer=zds4b1pt998">HISAT2</a></td>
        <td>A fast spliced aligner with low memory requirements</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Alignments in SAM/BAM format</td>
        <td><a href="https://www.nature.com/articles/nmeth.3317">Kim <em>et al</em>., 2015, Nature Methods</a></td>
   </tr>
   <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=bwa-mem&version=18.09&__identifer=eo9vnwpvp9">bwa-mem</a></td>
        <td>Fast and accurate short read alignment with Burrows–Wheeler transform</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Alignments in SAM/BAM format</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/25/14/1754/225615">Li <em>et al</em>., 2009, Bioinformatics</a></td>
   </tr>
 </table>
</div>

## CMR Calling
**CMR Calling** implements four pipelines for MeRIP-Seq, m6A miCLIP-Seq, CeU-Seq and RNA-BSSeq, respectively.

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Tools</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>Programs</b></td>
        <td><b>References</b></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Calling_peaks&version=18.09&__identifer=gz3la9l6aur">Peak calling from the MeRIP-Seq data</a></td>
        <td>Identifying enriched genomic regions from MeRIP-Seq experiment</td>
        <td>Alignments of IP and input in SAM/BAM format and Reference genome sequences in FASTA format</td>
        <td>CMR regions in BED format</td>
        <td><a href="https://github.com/cma2015/PEA">PEA</a></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/34/21/3747/5021690">Zhai <em>et al</em>., 2018, Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=peakcalling_m5C&version=18.09&__identifer=kgd6vgn38w">Calling m5C from the RNA-BSseq data</a></td>
        <td>Performing bisulfite sequencing (BS-Seq) read mapping, comprehensive methylation calling using meRanTK</td>
        <td>Sequencing reads in FASTQ format and reference genome sequences in FASTTA format</td>
        <td>m5C sites in BED format</td>
        <td><a href="http://icbi.at/software/meRanTK">meRanTK</a></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/32/5/782/1744216">Rieder <em>et al</em>., 2016, Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Calling_pseudoU&version=18.09&__identifer=nrfo2zxngp">Calling Ψ from CeU-Seq data</a></td>
        <td>Identifying pseudouridylation from CeU-Seq</td>
        <td>Alignments in SAM/BAM format and cDNA sequences in FASTA format</td>
        <td>Pseudoridylation sites in BED format</td>
        <td>In-house scripts</td>
        <td><a href="https://www.nature.com/articles/nchembio.1836">Li <em>et al</em>., 2015, Nature Chemical Biology</a></td>
    </tr>
</table>
</div>

## Machine Learning-based CMR Prediction
This sub-module provides a pipeline for transcriptome-wide CMR prediction using machine learning technology. This pipeline is consisted of [Feature encoding](#feature-encoding), [Feature selection](#feature-selection) and [Machine learning-based classification](#machine-learning-based-classification)

### Feature encoding

**Sequence-derived features**

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Type</b></td>
        <td><b>Feature</b></td>
        <td><b>Description</b></td>
        <td><b>Dimension</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td rowspan="4">Nucleic acid composition related features</td>
        <td>1mer</td>
        <td>The frequency of mono-nucleotide</td>
        <td>4</td>
        <td rowspan="4"><a href="https://academic.oup.com/bib/article-abstract/20/4/1280/4763667?redirectedFrom=fulltext">Liu <em>et al</em>., 2017, Briefings in Bioinformatics</a></td>
    </tr>
    <tr>
        <td>2mer</td>
        <td>The frequency of di-nucleotide</td>
        <td>16</td>
    </tr>
    <tr>
        <td>3mer</td>
        <td>The frequency of tri-nucleotide</td>
        <td>64</td>
    </tr>
    <tr>
        <td>4mer</td>
        <td>The frequency of tetra-nucleotide</td>
        <td>256</td>
    </tr>
    <tr>
        <td rowspan="6">Autocorrelation-base features</td>
        <td>DAC (<B><U>D</U></B>inucleotide-based <B><U>a</U></B>uto <B><U>c</U></B>ovariance)</td>
        <td>DAC measures the correlation of the same physicochemical index between two dinucleotides seperated by a distance of <I>d</I> along the sequence</td>
        <td><I>N</I>*<I>d</I></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/25/20/2655/194053">Dong <em>et al</em>., 2009, Bioinformatics</a> <a href="https://academic.oup.com/nar/article/36/9/3025/1104168">Guo <em>et al</em>., 2008, Nucleic Acids Research</a> <a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158">Chen <em>et al</em>., 2015, Bioinformatics</a></td>
    </tr>
    <tr>
        <td>DCC (<B><U>D</U></B>inucleotide-based <B><U>c</U></B>ross <B><U>c</U></B>ovariance)</td>
        <td>DCC measures the correlation of two different physicochemical indices between two dinucleotides separated by a distance of <I>d</I> nucleotide acids along the sequence</td>
        <td><I>N</I>*(<I>N-1</I>)*<I>d</I></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/25/20/2655/194053">Dong <em>et al</em>., 2009, Bioinformatics</a> <a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158">Chen <em>et al</em>., 2015, Bioinformatics</a></td>
    </tr>
    <tr>
        <td>DACC (<B><U>D</U></B>inucleotide-based <B><U>a</U></B>uto-<B><U>c</U></B>ross <B><U>c</U></B>ovariance)</td>
        <td>A combination of DAC and DCC</td>
        <td><I>N</I>*<I>N</I>*<I>d</I></td>
         <td><a href="https://academic.oup.com/bioinformatics/article/25/20/2655/194053">Dong <em>et al</em>., 2009, Bioinformatics</a> <a href="https://academic.oup.com/nar/article/36/9/3025/1104168">Guo <em>et al</em>., 2008, Nucleic Acids Research</a> <a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158">Chen <em>et al</em>., 2015, Bioinformatics</a></td>
    </tr>
    <tr>
        <td>MAC (<B><U>M</U></B>oran <B><U>a</U></B>uto<B><U>c</U></B>orrelation)</td>
        <td>MAC measures the correlation of the same properties between two residues separated by a distance of <I>d</I> along the sequence</td>
        <td><I>N</I>*<I>k</I>*<I>d</I></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158">Chen <em>et al</em>., 2015, Bioinformatics</a> <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/bip.360270308">Horne <em>et al</em>., 1988, Biopolymers</a></td>
    </tr>
    <tr>
        <td>GAC (<B><U>G</U></B>eary <B><U>a</U></B>uto<B><U>c</U></B>orrelation)</td>
        <td>GAC measures the correlation of the same properties between two residues separated by a distance of <I>d</I> along the sequence</td>
        <td><I>N</I>*<I>k</I>*<I>d</I></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158">Chen <em>et al</em>., 2015, Bioinformatics</a> <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ajpa.20250">Sokal <em>et al</em>., 2006, American
journal of physical anthropology</a></td>
    </tr>
    <tr>
        <td>NMBAC (<B><U>N</U></B>ormalized <B><U>M</U></B>oreau-<B><U>B</U></B>roto <B><U>a</U></B>uto<B><U>c</U></B>orrelation)</td>
        <td>It measures the corelatio of the same properties between two residues separated by a distance of <I>lag</I> along the sequence.</td>
        <td><I>N</I>*<I>d</I></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158">Chen <em>et al</em>., 2015, Bioinformatics</a> <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ajpa.20250">Feng <em>et al</em>., 2000, Journal of protein chemistry</a></td>
    </tr>
    <tr>
        <td rowspan="2">Pseudo nucleotide composition</td>
        <td>PC-PseDNC-General</td>
        <td>General parallel correlatio pseudo dinucleotide composition</td>
        <td>16+&lambda;</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158">Chen <em>et al</em>., 2015, Bioinformatics</a></td>
    </tr>
    <tr>
        <td>SC-PseDNC-General</td>
        <td>General series correlation pseudo dinucleotide composition</td>
        <td>16+&lambda;</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158">Chen <em>et al</em>., 2015, Bioinformatics</a></td>
    </tr>
    <tr>
        <td rowspan="2">Binary encoding</td>
        <td>Binary encoding</td>
        <td>A, C, G, U are encoded with (1, 0, 0, 0), (0, 1, 0, 0), (0, 0, 1, 0), (0, 0, 0, 1), respectively</td>
        <td>4*<I>L</I></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/34/21/3747/5021690">Zhai <em>et al</em>., 2018, Bioinformatics</a> <a href="https://www.frontiersin.org/articles/10.3389/fpls.2018.00519/full">Song <em>et al</em>., 2018, Frontiers in Plant Science</a></td>
    </tr>
    <tr>
        <td>Binary encoding based on structural chemical properties</td>
        <td>A, C, G, U are encoded as a vector of three features (1,1,1), (0,1,0), (1,0,0) and (0,0,1), respectively</td>
        <td>3*<I>L</I></td>
        <td><a href="https://academic.oup.com/nar/article/47/7/e41/5319125#133859200">Chen <em>et al</em>., 2019, Nucleic Acids Research</a></td>
    </tr>
</table>
</div>

**Note:**
- *N* is the number of physicochemical indices, here is 22;
- *L* is the sequence length;
- *d* is the length between two seperated nucleotides.
- <body><I>&lambda;</I></body> is an interger, representing the highest counted rank of the correlation along a RNA sequence.

**Genomic-derived features**
<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
     <tr>
        <td><b>Type</b></td>
        <td><b>Feature</b></td>
        <td><b>Description</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td rowspan="12">Dummy variables indicating whether the site is overlapped to the topological on the major RNA transcript</td>
        <td>Five prime UTR</td>
        <td>Binary value indicating whether CMR is overlapped with five prime UTR</td>
        <td rowspan="25"><a href="https://academic.oup.com/nar/article/47/7/e41/5319125#133859200">Chen <em>et al</em>., 2019, Nucleic Acids Research</a></td>
    </tr>
    <tr>
        <td>Three prime UTR</td>
        <td>Binary value indicating whether CMR is overlapped with three prime UTR</td>
    </tr>
    <tr>
        <td>Stop codons</td>
        <td>Binary value indicating whether CMR is overlapped with stop codons</td>
    </tr>
    <tr>
        <td>Start codons</td>
        <td>Binary value indicating whether CMR is overlapped with start codons</td>
    </tr>
    <tr>
        <td>Downstream 100bp of TSS (Transcription Start Sites)</td>
        <td>Binary value indicating whether CMR is located at the downstream 100bp of TSS</td>
    </tr>
    <tr>
        <td>Exons containing stop codons</td>
        <td>Binary value indicating whether stop codons are located at the exon containing CMR</td>
    </tr>
    <tr>
        <td>Alternative exons</td>
        <td>Binary value indicating the exon containing CMR is an alternative exon</td>
    </tr>
    <tr>
        <td>Constitutive exons</td>
        <td>Binary value indicating the exon containing CMR is a constitutive exon</td>
    </tr>
    <tr>
        <td>Internal exons</td>
        <td>Binary value indicating the exon containing CMR is an internal exon</td>
    </tr>
    <tr>
        <td>Long exons</td>
        <td>Binary value indicating the exon containing CMR is a long exon (exon length >= 400bp)</td>
    </tr>
    <tr>
        <td>Five prime 500bp of the last exon</td>
        <td>Binary value indicating whether the exon containing CMR is located at the five prime 500bp of the last exon</td>
    </tr>
    <tr>
        <td>Five prime 500bp of the last exon containing stop codons</td>
        <td>Binary value indicating whether the exon containing CMR is located at the five prime 500bp of the last exon containing stop codons</td>
    </tr>
    <tr>
        <td rowspan="3">Relative position on the region</td>
        <td>Relative position on five prime UTR</td>
        <td>An integer value indicating the relative position of CMR on five prime UTR</td>
    </tr>
    <tr>
        <td>Relative position on three prime UTR</td>
        <td>An integer value indicating the relative position of CMR on three prime UTR</td>
    </tr>
    <tr>
        <td>Relative position on exon</td>
        <td>An integer value indicating the relative position of CMR on the exon containing CMR</td>
    </tr>
    <tr>
        <td rowspan="3">The region length in bp</td>
        <td>The length of five prime UTR</td>
        <td>An interger value indicating the length of the five prime UTR containing CMR</td>
    </tr>
    <tr>
        <td>The length of three prime UTR</td>
        <td>An interger value indicating the length of three three prime UTR containing CMR</td>
    </tr>
    <tr>
        <td>Mature transcript length</td>
        <td>An interger value indicating the length of the mature transcript containing CMR</td>
    </tr>
    <tr>
        <td rowspan="3">Nucleotide distances toward the splicing junctions or the nearest neighboring sites</td>
        <td>Distance to the five prime splicing junction</td>
        <td>An interger indicating the distance to the five prime splicing junction</td>
    </tr>
    <tr>
        <td>Distance to the three prime splicing junction</td>
        <td>An interger indicating the distance to the three prime splicing junction</td>
    </tr>
    <tr>
        <td>Distance to the closest neighbor truncated at 200bp</td>
        <td>An interger indicating the distance to the closest neighbor truncated at 200bp</td>
    </tr>
    <tr>
        <td rowspan="4">Attributes of the genes of transcripts</td>
        <td>sncRNA</td>
        <td>Binary value indicating if the gene containing CMR is a sncRNA</td>
    </tr>
    <tr>
        <td>lncRNA</td>
        <td>Binary value indicating if the gene containing CMR is a lncRNA</td>
    </tr>
    <tr>
        <td>House keeping genes</td>
        <td>Binary value indicating if the gene containing CMR is a house keeping gene</td>
    </tr>
    <tr>
        <td>miRNA targeted genes</td>
        <td>Binary value indicating if the gene containing CMR is an miRNA target gene</td>
    </tr>
</table>
</div>

### Machine learning-based classification
In this module, several commonly-used machine learning classification algorithms are implemented to construct a CMR predictor. The detailes are described as follows:

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Algorithms</b></td>
        <td><b>Description</b></td>
        <td colspan="4"><b>Parameters</b></td>
    </tr>
    <tr>
        <td>Random Forest (RF)</td>
        <td>RF is an ensemble learning method for classification, regression and other tasks that operates by constructing a multitude of decision trees at training time and outputting the class that is the mode of the classes (classification) or mean prediction (regression) of the individual trees</td>
        <td colspan="4"><B>&#9312;</B> Number of trees to grow;<br> <B>&#9313;</B> The number of threads used for parallel computation.</td>
    </tr>
    <tr>
        <td>Support Vector Machine (SVM)</td>
        <td>Given a set of training examples, each marked as belonging to one or the other of two categories, an SVM training algorithm builds a model that assigns new examples to one category or the other, making it a non-probabilistic binary linear classifier (although methods such as Platt scaling exist to use SVM in a probabilistic classification setting)</td>
        <td>Polynomial kernel:<br><B>&#9312;</B> degree<br> <B>&#9313;</B> gamma<br><B>&#9314;</B> coef0</td>
        <td>Radial basis kernel:<br> gamma<br></td>
        <td>Sigmoid kernel:<br> gamma<br></td>
        <td>Linear kernel, no required parameters</td>
    </tr>
    <tr>
        <td>Decision Tree</td>
        <td>A decision tree is a decision support tool that uses a tree-like model of decisions and their possible consequences, including chance event outcomes, resource costs, and utility. It is one way to display an algorithm that only contains conditional control statements</td>
        <td colspan="4"><B>&#9312;</B> Number of trees to grow;<br> <B>&#9313;</B> The number of threads used for parallel computation.</td>
    </tr>
    <tr>
        <td>XGBoost</td>
        <td>It provides a gradient boosting framework for classification</td>
        <td colspan="4"><B>&#9312;</B> Learning rate<br> <B>&#9313;</B> Number of trees to grow<br> <B>&#9314;</B> The number of threads used for parallel computation<br></td>
    </tr>
    <tr>
        <td>Logistic Regression</td>
        <td>In statistics, the logistic model (or logit model) is used to model the probability of a certain class or event existing such as pass/fail, win/lose, alive/dead or healthy/sick</td>
        <td colspan="4"><B>&#9312;</B> Optimization algorithm<br> <B>&#9313;</B> The number of threads used for parallel computation<br></td>
    </tr>
</table>
</div>

# Advanced-analysis
This module provides comprehensive CMR annotation and interactive visualization, both HTML format with interactive figures and tables and PDF format with publication-ready figures are available to analyze the distribution of CMRs in the genome and transcriptome.

## CMR Annotation and Visualization

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Functions</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>References</b></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=CMR_distribution&version=18.09&__identifer=328vozxeap">CMRs Distribution</a></td>
        <td>The distribution of CMRs in the genome and transcriptome including the number of peaks in genomic features, the regions of enrichment of CMRs within transcripts, the enrichment of CMRs in transcriptional start/stop site and the enrichment of CMRs in splicing sites;</td>
        <td><B>&#9312;</B> CMR regions in BED format<br> <B>&#9313;</B>  Genome annotation in GTF/GFF format</td>
        <td>Comprehensive overview of CMRs distribution in HTML or PDF format</td>
        <td>In-house scripts</td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=De_novo_motif_discovery&version=18.09&__identifer=rod01zsjan">De-novo Motif Discovery</a></td>
        <td>This function integrates MEME-ChIP, DREME and HOMER to perform <I>de-novo</I> motif discovery</td>
        <td>Positive sequences in FASTA format</td>
        <td>The discovered motifs in HTML format</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/27/12/1653/257754">Timothy <I>et al</I>., 2011, Bioinformatics</a><br><a href="https://academic.oup.com/bioinformatics/article/27/12/1696/255896">Philip <I>et al</I>., 2011, Bioinformatics</a><br><a href="https://doi.org/10.1016/j.molcel.2010.05.004">Heinz <I>et al</I>., 2010, Molecular Cell</a></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Motif_enrichment&version=18.09&__identifer=wwnbsugl6yd">Motif Enrichment</a></td>
        <td>This function integrates AME and CentriMo to perform motif enrichment</td>
        <td><B>&#9312;</B> Sequence file in FASTA format<br><B>&#9313;</B> Motif file in MEME format</td>
        <td>The motif enrichment in HTML format</td>
        <td><a href="http://www.biomedcentral.com/1471-2105/11/165">Robert <I>et al</I>., 2010, BMC Bioinformatics</a><br> <a href="https://academic.oup.com/nar/article/40/17/e128/2411117">Timothy <I>et al</I>., 2012, Nucleic Acids Research</a><br></td>
    </tr>
    <tr>
        <td><a href = "http://bioinfo.nwafu.edu.cn:4006/?tool_id=Sequence_logo_visualization&version=18.09&__identifer=ci3ahyxsn6p">Sequence Logo Visualization</a></td>
        <td>Visualize motifs in sequences</td>
        <td>Sequences in FASTA format</td>
        <td>Motif logo in PDF format</td>
        <td>In-house scripts</td>
    </tr>
    <tr>
        <td><a href = "http://bioinfo.nwafu.edu.cn:4006/?tool_id=Functional_enrichment_analysis&version=18.09&__identifer=y44oe05v30i">Functional Enrichment Analysis</a></td>
        <td>Perform GO or KEGG enrichment analysis for any species</td>
        <td>Gene list</td>
        <td>The enriched GO/KEGG terms</td>
        <td><a href = "https://doi.org/10.1089/omi.2011.0118">Yu <I>et al</I>., 2012, OMICS</a></td>
    </tr>
</table>
</div>


## Multi-omics Integration Analysis
This functions is designed to correlate CMR-realted gene with different type of genomic features including **Gene length analysis**, **Exon length analysis**, **Intron length analysis**, **Exon number analysis** and **GC content analysis**. To be specific, we split the  genome in a sliding window of 100 (default) adjacent genes with step size of 10 (default), and calculate the frequency of CMR genes in each window (100 adjacent genes). v2

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Genomic features</b></td>
        <td><b>Correlation</b></td>
        <td><b>Difference</b></td>
    </tr>
    <tr>
        <td>Gene length analysis</td>
        <td><img align="left" src="assets/img/Correlation_CMR_gene_length.png" alt="Correlation of CMR-related genes with gene length analysis" width="75px" valign="middle"></td>
        <td><img align="left" src="assets/img/Difference_CMR_gene_length.png" alt="Difference of CMR-related genes with gene length analysis" width="75px" valign="middle"></td>
    </tr>
    <tr>
        <td>Exon length analysis</td>
        <td><img align="left" src="assets/img/Correlation_CMR_exon_length.png" alt="Correlation of CMR-related genes with exon length analysis" width="75px" valign="middle"></td>
        <td><img align="left" src="assets/img/Difference_CMR_exon_length.png" alt="Difference of CMR-related genes with exon length analysis" width="75px" valign="middle"></td>
    </tr>
    <tr>
        <td>Intron length analysis</td>
        <td><img align="left" src="assets/img/Correlation_CMR_intron_length.png" alt="Correlation of CMR-related genes with intron length analysis" width="75px" valign="middle"></td>
        <td><img align="left" src="assets/img/Difference_CMR_intron_length.png" alt="Difference of CMR-related genes with intron length analysis" width="75px" valign="middle"></td>
    </tr>
    <tr>
        <td>Exon number analysis</td>
        <td><img align="left" src="assets/img/Correlation_CMR_exon_number.png" alt="Correlation of CMR-related genes with exon number analysis" width="75px" valign="middle"></td>
        <td><img align="left" src="assets/img/Difference_CMR_exon_number.png" alt="Difference of CMR-related genes with exon number analysis" width="75px" valign="middle"></td>
    </tr>
    <tr>
        <td>GC content analysis</td>
        <td><img align="left" src="assets/img/Correlation_CMR_GC_content.png" alt="Correlation of CMR-related genes with GC content analysis" width="75px" valign="middle"></td>
        <td><img align="left" src="assets/img/Difference_CMR_GC_content.png" alt="Difference of CMR-related genes with GC content analysis" width="75px" valign="middle"></td>
    </tr>
</table>
</div>


# Examples
- [CMRs distribution](http://cma2015.github.io/DeepEA/demo_output/CMR_distribution_single_case.html)
- [De-novo motif discovery](http://cma2015.github.io/DeepEA/demo_output/de-novo_motif_discovery.html)
- [Correlate CMR with gene length](http://cma2015.github.io/DeepEA/demo_output/CMR_genomic_features_for_gene_length.html)
- [Correlate CMR with exon length](http://cma2015.github.io/DeepEA/demo_output/CMR_genomic_features_for_exon_length.html)
- [Correlate CMR with intron length](http://cma2015.github.io/DeepEA/demo_output/CMR_genomic_features_for_intron_length.html)
- [Correlate CMR with exon number](http://cma2015.github.io/DeepEA/demo_output/CMR_genomic_features_for_exon_number.html)
- [Correlate CMR with GC content](http://cma2015.github.io/DeepEA/demo_output/CMR_genomic_features_for_GC_content.html)

# How to access help
* If users encounter any bugs or issues, feel free to leave a message at Github [issues](<https://github.com/cma2015/DeepEA/issues>). We will try our best to deal with all issues as soon as possible.
* In addition, if any suggestions are available, feel free to contact: Jingjing Zhai <zhaijingjing603@gmail.com> or Chuang Ma <chuangma2006@gmail.com>
