---
layout: default
---

# About
![docker pulls](https://img.shields.io/docker/pulls/malab/pea.svg)
![docker stars](https://img.shields.io/docker/stars/malab/pea.svg)

DeepEA is a Galaxy-based framework for exploring large-scale analysis of epitranscriptome sequencing data. DeepEA is an upgrated version of PEA, which is a special designed R toolkit for plant epitranscriptome analysis. In the current version of DeepEA, a variety of modules and functions including Data Preparation, Quality Control, CMR (chemical modifications of RNA) profiling and analysis, CMR annotation, CMR prediction and related visualization are implemented to enable users perform large-scale, reproducible epitranscriptome analysis. In addition, by taking advantages of multiple instance learning (MIL) algorithms, DeepEA can precisely localize CMRs in peak regions. This can reduce the false positive CMRs generated by MeRIP-Seq/m6A-Seq/m5C-RIP-Seq, as which can localize CMRs to transcript regions of 100-200 nucleotide long but fail to accurately identify CMRs at single-nucleotide solution. The deePEA project is hosted on https://github.com/cma2015/DeepEA. The DeepEA web server can accessed via http://bioinfo.nwafu.edu.cn:4006.

<div id="top"></div>

[<img align="left" src="assets/img/deNBI_logo.jpg" height="35px" alt="de.NBI" valign="middle">](http://www.denbi.de)
[<img align="center" src="assets/img/elixir_germany.png" height="55px" alt="ELIXIR Germany" valign="middle">](https://www.elixir-europe.org)

- [Installation](#installation)
	- [Docker installation](#docker-installation)
	- [DeepEA quick start](#deepea-quick-start)
- [Pre-analysis](pre-analysis)
	- [Data Preparation](#data-preparation)
	- [Quality control](#quality-control)
- [Core-analysis](#core-analysis)
	- [CMR Profiling](#cmr-profiling)
		- [Align Reads to Genome](#align-reads-to-genome)
		- [CMR Calling](#cmr-calling)
		- [Detect Differential CMR Regions](#detect-differential-cmr-regions)
    - [Machine Learning-based CMR Prediction](#machine-learning-based-cmr-prediction)
        - [Feature encoding](#feature-encoding)
        - [Feature selection](#feature-selection)
        - [Machine learning-based classification](#machine-learning-based-classification)
- [Advanced-analysis](#advanced-analysis)
	- [CMR Annotation and Visualization](#cmr-annotation-and-visualization)
		- [Distribution Analysis](#distribution-analysis)
		- [Motif Analysis](#motif-analysis)
		- [Sequence Logo Visualization](#sequence-logo-visualization)
		- [Functional Enrichment Analysis](#functional-enrichment-analysis)
    - [Multi-omics Integration Analysis](#multi-omics-integration-analysis)
- [How to access help](#how-to-access-help)

# Installation
## Docker installation

> Official installation tutorial is available at [here](https://docs.docker.com/install)

### For Windows (Test on Windows 10 Enterprise version):
* Download [Docker](<https://download.docker.com/win/stable/Docker%20for%20Windows%20Installer.exe>) for windows </br>
* Double click the EXE file to open it;
* Follow the wizard instruction and complete installation;
* Search docker, select ___Docker for Windows___ in the search results and clickit.

### For Mac OS X (Test on macOS Sierra version 10.12.6 and macOS High Sierra version 10.13.3):
* Download [Docker](<https://download.docker.com/mac/stable/Docker.dmg>) for Mac OS X
* Double click the DMG file to open it;
* Drag the docker into Applications and complete installation;
* Start docker from Launchpad by click it.

### For Ubuntu (Test on Ubuntu 16.04 LTS):

```bash
# Update the apt package index:
sudo apt-get update
# Install packages to allow apt to use a repository over HTTPS:
sudo apt-get install \
    apt-transport-https \
    ca-certificates \
    curl \
    software-properties-common
# Add Dockerâ€™s official GPG key
curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
sudo apt-key fingerprint 0EBFCD88
sudo add-apt-repository \
   "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
   $(lsb_release -cs) \
   stable"
sudo apt-get update
sudo apt-get install docker-
# Verify that Docker CE is installed correctly by running the hello-world image
sudo docker run hello-world
```

### Verify if Docker is installed correctly

Once Docker installation is completed, we can run ____hello-world____ image to verify if Docker is installed correctly. Open terminal in Mac OS X and Linux operating system and open CMD for Windows operating system, then type the following command:

```bash
$ docker run hello-world
```
**<font color =red>Note</font>:** root permission is required for Linux operating system, if you want to manage docker as a non-root user, please refer [here](https://docs.docker.com/install/linux/linux-postinstall)



## DeepEA quick start

DeepEA has been packaged into a docker image, which can be easily installed as a local server by following command.

```bash
# pull DeepEA from docker hub
docker pull malab/deepea
docker run -it malab/deepea /bin/bash
bash run.sh
```

Then DeepEA local server can be accessed via `http://localhost:8080`

# Pre-analysis

**Pre-analysis** module is composed of two sub-modules: [Data Preparation](#data-preparation) and [Quality Control](#quality-control). These two sub-modules make up the important components of Epitranscriptome.

## Data Preparation

This sub-module provides three funcitons (see following table for details) to prepare epitranscriptome data.
<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Tools</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>References</b></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Obtain_Genome_or_Transcriptome_Sequences_and_Annotation&version=18.09&__identifer=dclshuq6eet">Obtain Genome Sequences and Annotation</a></td>
        <td>Fetch genome, cDNA, CDS sequences and annotation from Enseml automatically</td>
        <td>Select an Ensembl database; Select a species; Select dada type</td>
        <td>FASTA format sequences or GTF annotation</td>
        <td>In-house scripts</td>
   </tr>
   <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Obtain_Epitranscriptome_Sequencing_Reads&version=18.09&__identifer=3bj51oj9isq">Obtain Epitranscriptome Sequencing and Reads</a></td>
        <td>Fetch epitranscriptome sequencing reads from NCBI's SRA database</td>
        <td>SRR accession or HTTP/FTP link</td>
        <td>Sequencing files in SRA/FASTQ format</td>
        <td><a href="https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft">SRA Toolkit</a></td>
   </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Convert_Format&version=18.09&__identifer=uib940h4pb">Sequence Data Preprocessing</a></td>
        <td>xxx</td>
        <td>xxx</td>
        <td>xxx</td>
        <td>xxx</td>
   </tr>
 </table>
</div>

## Quality Control
**Quality Control** submodule consisted of a suite of tools focused on different levels of quality assessment including reads quality, alignment quality and CMR quality, Thus, three functions are implemented including **Assess Reads Quality**, **Assess Alignment Quality** and **Assess CMR quality**. The details of as follows:

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Tool</b></td>
        <td><b>Description</b></td>
        <td><b>Main functions</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>Programs</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td><a href='http://bioinfo.nwafu.edu.cn:4006/?tool_id=Assess_Reads_Quality&amp;version=18.09&amp;__identifer=gr96n8z0sag'>Assess Reads Quality</a></td>
        <td>This tool integrates FastQC and fastp to perform quality control from raw epitranscriptome sequencing reads</td>
        <td>Reads trimming and generate reads quality report including sequence quality, GC content, etc</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Clean reads or reads quality report in HTML format</td>
        <td><a href='https://www.bioinformatics.babraham.ac.uk/projects/fastqc/'>FastQC</a>,<a href='https://github.com/OpenGene/fastp'>fastp</a></td>
        <td><a href='https://doi.org/10.1093/bioinformatics/bty560'>Chen <em>et al</em>., 2018, Bioinformatics</a>, <a href='http://www.bioinformatics.babraham.ac.uk/projects/fastqc'>Babraham Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href='http://bioinfo.nwafu.edu.cn:4006/?tool_id=Assess_Alignment_Quality&amp;version=18.01&amp;__identifer=3uvvorrbx2s'>Assess Alignment Quality</a></td>
        <td>This function is used to generate quality assessment.</td>
        <td>Evaluate alignment quality</td>
        <td>Reads alignment fies in SAM/BAM format</td>
        <td>Alignment quality report in HTML format</td>
        <td><a href='https://github.com/skyhorsetomoon/Trumpet'>trumpet</a></td>
        <td><a href='https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2266-3'>Zhang et al., 2018, BMC Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href='http://bioinfo.nwafu.edu.cn:4006/?tool_id=Assess_CMR_Regions_Quality&amp;version=18.01&amp;__identifer=yu0s7b4tvw7'>Assess CMR Quality</a></td>
        <td>This tool aims to quantify CMR signal strength by counting reads and calculating RPKM in binding site intervals</td>
        <td>Quantify CMRs</td>
        <td>Alignments in SAM/BAM format</td>
        <td>CMR quantification matrix</td>
        <td><a href='https://bioconductor.org/packages/release/bioc/html/DiffBind.html'>DiffBind</a></td>
        <td><a href='https://www.frontiersin.org/articles/10.3389/fgene.2015.00169/full'>Wu <em>et al</em>., 2016, frontiers in Genetics</a></td>
    </tr>
</table>
</div>

# Core-analysis
**Core-analysis** module consists of **CMR Profiling** and **Machine Learning-based CMR Prediction**, of which **CMR Profiling** provides step-by-step functions required for epitranscriptome reads mapping, CMR Calling and differential CMR analysis, while **Machine Learning-based CMR Prediction** is designed to automatically complete the three main steps for constructing a CMR predictor including **Sequence Feature Extraction**, **Feature Engineering** and **Machine Learning-based Classification**

## Align Reads to Genome
Several commonly used aligners are wrapped to align epitranscriptome reads to genome. Currently, [Tophat2](https://ccb.jhu.edu/software/tophat/index.shtml), [Bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml), [STAR](https://github.com/alexdobin/STAR), [HISAT2](http://www.ccb.jhu.edu/software/hisat/index.shtml) and [bwa-mem](http://bio-bwa.sourceforge.net/bwa.shtml) are available.

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Tools</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>References</b></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=tophat2&version=18.09&__identifer=jztsfk48b9">Tophat2</a></td>
        <td>Accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Alignments in SAM/BAM format</td>
        <td><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36">Kim <em>et al</em>., 2013, Genome Biology</a></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=bowtie2&version=18.09&__identifer=x9j53etnvzb">Bowtie2</a></td>
        <td>A fast gapped-read alignment tool</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Alignments in SAM/BAM format</td>
        <td><a href="https://www.nature.com/articles/nmeth.1923">Langmead <em>et al</em>., 2012, Nature Methods</a></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=star&version=18.09&__identifer=gd64qwrc1pp">STAR</a></td>
        <td>An ultrafast universal RNA-seq aligner</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Alignments in SAM/BAM format</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/29/1/15/272537">Dobin <em>et al</em>., 2013, Bioinformatics</a></td>
   </tr>
   <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=hisat2&version=18.09&__identifer=zds4b1pt998">HISAT2</a></td>
        <td>A fast spliced aligner with low memory requirements</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Alignments in SAM/BAM format</td>
        <td><a href="https://www.nature.com/articles/nmeth.3317">Kim <em>et al</em>., 2015, Nature Methods</a></td>
   </tr>
   <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=bwa-mem&version=18.09&__identifer=eo9vnwpvp9">bwa-mem</a></td>
        <td>Fast and accurate short read alignment with Burrowsâ€“Wheeler transform</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Alignments in SAM/BAM format</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/25/14/1754/225615">Li <em>et al</em>., 2009, Bioinformatics</a></td>
   </tr>
 </table>
</div>

## CMR Calling
**CMR Calling** implements four pipelines for MeRIP-Seq, m6A miCLIP-Seq, CeU-Seq and RNA-BSSeq, respectively.

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Tools</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>Programs</b></td>
        <td><b>References</b></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Calling_peaks&version=18.09&__identifer=gz3la9l6aur">Peak calling from the MeRIP-Seq data</a></td>
        <td>Identifying enriched genomic regions from MeRIP-Seq experiment</td>
        <td>Alignments of IP and input in SAM/BAM format and Reference genome sequences in FASTA format</td>
        <td>CMR regions in BED format</td>
        <td><a href="https://github.com/cma2015/PEA">PEA</a></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/34/21/3747/5021690">Zhai <em>et al</em>., 2018, Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=peakcalling_m5C&version=18.09&__identifer=kgd6vgn38w">Calling m5C from the RNA-BSseq data</a></td>
        <td>Performing bisulfite sequencing (BS-Seq) read mapping, comprehensive methylation calling using meRanTK</td>
        <td>Sequencing reads in FASTQ format and reference genome sequences in FASTTA format</td>
        <td>m5C sites in BED format</td>
        <td><a href="http://icbi.at/software/meRanTK">meRanTK</a></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/32/5/782/1744216">Rieder <em>et al</em>., 2016, Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href="http://bioinfo.nwafu.edu.cn:4006/?tool_id=Calling_pseudoU&version=18.09&__identifer=nrfo2zxngp">Calling Î¨ from CeU-Seq data</a></td>
        <td>Identifying pseudouridylation from CeU-Seq</td>
        <td>Alignments in SAM/BAM format and cDNA sequences in FASTA format</td>
        <td>Pseudoridylation sites in BED format</td>
        <td>In-house scripts</td>
        <td><a href="https://www.nature.com/articles/nchembio.1836">Li <em>et al</em>., 2015, Nature Chemical Biology</a></td>
    </tr>
</table>
</div>

## Machine Learning-based CMR Prediction
This sub-module provides a pipeline for transcriptome-wide CMR prediction using machine learning technology. This pipeline is consisted of [Feature encoding](#feature-encoding), [Feature selection](#feature-selection) and [Machine learning-based classification](#machine-learning-based-classification)

### Feature encoding

- **Sequence-derived features**

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
    <tr>
        <td><b>Type</b></td>
        <td><b>Feature</b></td>
        <td><b>Description</b></td>
        <td><b>Dimension</b></td>
    </tr>
    <tr>
        <td rowspan="4">Nucleic acid composition related features</td>
        <td>1mer</td>
        <td>The frequency of mono-nucleotide</td>
        <td>4</td>
    </tr>
    <tr>
        <td>2mer</td>
        <td>The frequency of di-nucleotide</td>
        <td>16</td>
    </tr>
    <tr>
        <td>3mer</td>
        <td>The frequency of tri-nucleotide</td>
        <td>64</td>
    </tr>
    <tr>
        <td>4mer</td>
        <td>The frequency of tetra-nucleotide</td>
        <td>256</td>
    </tr>
    <tr>
        <td rowspan="6">Autocorrelation-base features</td>
        <td>DAC (<B><U>D</U></B>inucleotide-based <B><U>a</U></B>uto <B><U>c</U></B>ovariance)</td>
        <td>xxx</td>
        <td>xx</td>
    </tr>
    <tr>
        <td>DCC (<B><U>D</U></B>inucleotide-based <B><U>c</U></B>ross <B><U>c</U></B>ovariance)</td>
        <td>xxx</td>
        <td>xxx</td>
    </tr>
    <tr>
        <td>DACC (<B><U>D</U></B>inucleotide-based <B><U>a</U></B>uto-<B><U>c</U></B>ross <B><U>c</U></B>ovariance)</td>
        <td>A combination of DAC and DCC</td>
        <td>xxx</td>
    </tr>
    <tr>
        <td>MAC (<B><U>M</U></B>oran <B><U>a</U></B>uto<B><U>c</U></B>orrelation)</td>
        <td>xxx</td>
        <td>xxx</td>
    </tr>
    <tr>
        <td>GAC (<B><U>G</U></B>eary <B><U>a</U></B>uto<B><U>c</U></B>orrelation)</td>
        <td>xxx</td>
        <td>xxx</td>
    </tr>
    <tr>
        <td>NMBAC (<B><U>N</U></B>ormalized <B><U>M</U></B>oreau-<B><U>B</U></B>roto <B><U>a</U></B>uto<B><U>c</U></B>orrelation)</td>
        <td>It measures the corelatio of the same properties between two residues separated by a distance of <I>lag</I> along the sequence.</td>
        <td>xxx</td>
    </tr>
    <tr>
        <td rowspan="2">Pseudo nucleotide composition</td>
        <td>PC-PseDNC-General</td>
        <td>x</td>
        <td>x</td>
    </tr>
    <tr>
        <td>SC-PseDNC-General</td>
        <td>x</td>
        <td>x</td>
    </tr>
    <tr>
        <td rowspan="2">Binary encoding</td>
        <td>Binary encoding</td>
        <td>A, C, G, U are encoded with (1, 0, 0, 0), (0, 1, 0, 0), (0, 0, 1, 0), (0, 0, 0, 1), respectively</td>
        <td>4*<I>L</I></td>
    </tr>
    <tr>
        <td>Binary encoding based on structural chemical properties</td>
        <td>A, C, G, U are encoded as a vector of three features (1,1,1), (0,1,0), (1,0,0) and (0,0,1), respectively</td>
        <td>3*<I>L</I></td>
    </tr>
</table>
</div>

Genomic-derived features v2

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="">
     <tr>
        <td><b>Type</b></td>
        <td><b>Feature</b></td>
        <td><b>Description</b></td>
        <td><b>Dimension</b></td>
    </tr>
    <tr>
        <td>x</td>
        <td>x</td>
        <td>x</td>
        <td>x</td>
    </tr>
</table>
</div>

### Machine learning-based classification

xxx

# Advanced-analysis


# How to access help
* If users encounter any bugs or issues, feel free to leave a message at Github [issues](<https://github.com/cma2015/DeepEA/issues>). We will try our best to deal with all issues as soon as possible.
* In addition, if any suggestions are available, feel free to contact: Jingjing Zhai <zhaijingjing603@gmail.com> or Chuang Ma <chuangma2006@gmail.com>
