---
layout: default
---

# About
![docker pulls](https://img.shields.io/docker/pulls/malab/pea.svg)
![docker stars](https://img.shields.io/docker/stars/bgruening/pea.svg)

DeepEA is a Galaxy-based framework for exploring large-scale analysis of epitranscriptome sequencing data. DeepEA is an upgrated version of PEA, which is a special designed R toolkit for plant epitranscriptome analysis. In the current version of DeepEA, a variety of modules and functions including Data Preparation, Quality Control, CMR (chemical modifications of RNA) profiling and analysis, CMR annotation, CMR prediction and related visualization are implemented to enable users perform large-scale, reproducible epitranscriptome analysis. In addition, by taking advantages of multiple instance learning (MIL) algorithms, DeepEA can precisely localize CMRs in peak regions. This can reduce the false positive CMRs generated by MeRIP-Seq/m6A-Seq/m5C-RIP-Seq, as which can localize CMRs to transcript regions of 100-200 nucleotide long but fail to accurately identify CMRs at single-nucleotide solution. The deePEA project is hosted on https://github.com/cma2015/DeepEA. The DeepEA web server can accessed via http://bioinfo.nwafu.edu.cn:4006.

<div id="top"></div>

[<img align="left" src="assets/img/deNBI_logo.jpg" height="35px" alt="de.NBI" valign="middle">](http://www.denbi.de)
[<img align="center" src="assets/img/elixir_germany.png" height="55px" alt="ELIXIR Germany" valign="middle">](https://www.elixir-europe.org)

- [Installation](#installation)
	- [Docker installation](#docker-installation)
	- [DeepEA quick start](#DeepEA-quick-start)
- [Pre-analysis](#Pre-analysis)
	- [Data Preparation](#Data-preparation)
		- [Obtain Genome Sequences and Annotation](#Obtain-genome-sequences-and-annotation)
		- [Obtain Epitranscriptome Sequencing Reads](#Obtain-epitranscriptome-sequencing-reads)
		- [Convert format](#Convert-format)
	- [Quality control](#Quality-control)
		- [Assess Reads Quality](#Assess-read-quality)
		- [Assess Alignment Quality](#Assess-alignment-quality)
		- [Assess CMR Regions Quality](#Assess-CMR-regions-quality)
- [Core-analysis](#Core-analysis)
	- [CMR Profiling](#CMR-profiling)
		- [Align Reads to Genome](#Align-reads-to-genome)
			- [Tophat2](#Tophat2)
			- [Bowtie2](#Bowtie2)
			- [STAR](#STAR)
			- [HISAT2](#HISAT2)
		- [CMR Calling](#CMR-calling)
			- [Peak Calling from the MeRIP-Seq data](#Peak-calling-from-the-MeRIP-Seq-data)
			- [Calling m5C from the RNA-BSeq data](#Calling-m5C-from-the-RNA-BSeq-data)
			- [CMR Calling from the miCLIP-Seq data](#Calling-m5C-from-the-miCLIP-Seq-data)
			- [Calling pseudouridine from CeU-Seq data](#Calling-pseudouridine-from-CeU-Seq-data)
		- [Merge Biological Replicates](#Merge-biological-replicates)
	- [CMR Annotation and Visualization](#CMR-annotations-and-visualization)
		- [Distribution Analysis](#Distribution-analysis)
		- [Motif Analysis](#Motif-analysis)
		- [Sequence Logo Visualization](#Sequence-logo-visualization)
		- [Functional Enrichment Analysis](#Functional-enrichment-analysis)
- [Advanced Analysis](#Advanced-analysis)
- [How to access help](#how-to-access-help)
- [Contact](#Contact)

# Installation
