---
layout: default
---

# Brief introduction
![docker pulls](https://img.shields.io/docker/pulls/malab/pea.svg)
![docker stars](https://img.shields.io/docker/stars/malab/pea.svg)

<div id="top"></div>
DeepEA is a Galaxy-based framework for exploration and large-scale analysis of epitranscriptome sequencing data. DeepEA is an upgrated version of [PEA](https://github.com/cma2015/PEA), which is a special designed R toolkit for plant epitranscriptome analysis. In the current version of DeepEA, a variety of modules and functions including data preparation, quality control, CMR (chemical modifications of RNA) profiling and analysis, CMR annotation, CMR prediction and related visualization are implemented to enable users perform large-scale, reproducible epitranscriptome analysis. In addition, by taking advantages of multiple instance learning (MIL) algorithms, DeepEA can precisely localize CMRs in peak regions. This can reduce the false positive CMRs generated by MeRIP-Seq/m6A-Seq/m5C-RIP-Seq, as which can localize CMRs to transcript regions of 100-200 nucleotide long but fail to accurately identify CMRs at single-nucleotide solution. The deePEA project is hosted on https://github.com/cma2015/DeepEA. The DeepEA web server can accessed via http://bioinfo.nwafu.edu.cn:4006.


[<img align="left" src="assets/img/malab.png" height="35px" alt="malab" valign="middle">](http://www.bioinfo.nwafu.edu.cn)

- [Installation](#installation)
  - [Docker installation](#docker-installation)
  - [DeepEA quick start](#DeepEA-quick-start)
- [Pre-analysis](#Pre-analysis)
  - [Data Preparation](#Data-preparation)
    - [Obtain Genome Sequences and Annotation](Obtain-genome-sequences-and-annotation)
    - [Obtain Epitranscriptome Sequencing Reads](Obtain-epitranscriptome-sequencing-reads)
    - [Convert format](Convert-format)
  - [Quality control](#Quality-control)
    - [Assess Reads Quality](#Assess-read-quality)
    - [Assess Alignment Quality](#Assess-alignment-quality)
    - [Assess CMR Regions Quality](#Assess-CMR-regions-quality)
- [How to access help](#how-to-access-help)


# Installation
## Docker installation ([**Official installation tutorial**](https://docs.docker.com/install))

### For Windows (Test on Windows 10 Enterprise version):
* Download [Docker](<https://download.docker.com/win/stable/Docker%20for%20Windows%20Installer.exe>) for windows </br>
* Double click the EXE file to open it;
* Follow the wizard instruction and complete installation;
* Search docker, select ___Docker for Windows___ in the search results and clickit.
#### For Mac OS X (Test on macOS Sierra version 10.12.6 and macOS High Sierra version 10.13.3):
* Download [Docker](<https://download.docker.com/mac/stable/Docker.dmg>) for Mac os <br>
* Double click the DMG file to open it;
* Drag the docker into Applications and complete installation;
* Start docker from Launchpad by click it.
#### For Ubuntu (Test on Ubuntu 14.04 LTS and Ubuntu 16.04 LTS):
* Go to [Docker](<https://download.docker.com/linux/ubuntu/dists/>), choose your Ubuntu version, browse to ___pool/stable___ and choose ___amd64, armhf, ppc64el or s390x.____ Download the ___DEB___ file for the Docker version you want to install;
* Install Docker, supposing that the DEB file is download into following path:___"/home/docker-ce<version->~ubuntu_amd64.deb"___ </br>
```bash
$ sudo dpkg -i /home/docker-ce<version-XXX>~ubuntu_amd64.deb      
$ sudo apt-get install -f
```

### Verify if Docker is installed correctly

Once Docker installation is completed, we can run ____hello-world____ image to verify if Docker is installed correctly. Open terminal in Mac OS X and Linux operating system and open CMD for Windows operating system, then type the following command:

```bash
$ docker run hello-world
```
**<font color =red>Note</font>:** root permission is required for Linux operating system.

## DeepEA quick start

```bash
docker pull malab/DeepEA
```

# Pre-analysis

# How to access help