<!DOCTYPE html>
<html>
	<head>

    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="DeepEA: a Galaxy-based framework for exploration and large-scale analysis of epitranscriptome sequencing data">
    <meta name="author" content="The DeepEA Team">

    <title>DeepEA</title>

    <!-- Bootstrap Core CSS -->
    <link href="assets/css/bootstrap.css" rel="stylesheet">

    <!-- Custom CSS -->
    <link href="assets/css/landing-page.css" rel="stylesheet">
	<link href="assets/css/canvas.css" rel="stylesheet">
	<link href="assets/css/lunbo.css" rel="stylesheet">
	<link href="assets/css/button.css" rel="stylesheet">
	<link rel="stylesheet" type="text/css" href="assets/css/alert.css">
	<script src="assets/js/lunbo.js"></script>
	<script src="assets/js/table.js"></script>
    <link href="https://fonts.googleapis.com/css?family=Lato:300,400,700,300italic,400italic,700italic" rel="stylesheet" type="text/css">

</head>

	<body  onload="init(70)">
	<canvas id="canvas"></canvas>
		<script>
			window.requestAnimationFrame = window.requestAnimationFrame
										|| window.mozRequestAnimationFrame
										|| window.webkitRequestAnimationFrame
										|| window.msRequestAnimationFrame;

			var canvas = document.getElementById('canvas');
			var ctx = canvas.getContext('2d');

			var w = canvas.width = canvas.offsetWidth;
			var h = canvas.height = canvas.offsetHeight;
			var circles = [];
			var current_circle = new currentCircle(0, 0);

			function init(num) {
				for(var i = 0; i < num; i ++) {
					circles.push(new circle(Math.random() * w, Math.random() * h));
				}
				draw();
			}

			function draw() {
				ctx.clearRect(0, 0, w, h);
				for(var i = 0; i < circles.length; i ++) {
					circles[i].move(w, h);
					circles[i].drawCircle(ctx);
					for(var j = i + 1; j < circles.length; j++) {
						circles[i].drawLine(ctx, circles[j]);
					}
				}
				if (current_circle.x) {
					current_circle.drawCircle(ctx);
					for (var k = 0; k < circles.length; k++) {
						current_circle.drawLine(ctx, circles[k])
					}
				}
				requestAnimationFrame(draw);
			}



			function circle(x, y) {
				this.x = x;
				this.y = y;
				this.r = Math.random() * 10;
				this._mx = 1 - (Math.random() * 2);
				this._my = 1 - (Math.random() * 2);

				this.drawCircle = drawCircle;
				function drawCircle(ctx) {
					ctx.beginPath();
					ctx.arc(this.x, this.y, this.r, 0, 360);
					ctx.fillStyle = 'rgba(204, 204, 204, 0.3)';
					ctx.fill();
				}

				this.move = move;
				function move(w, h) {
					this._mx = (this.x < w && this.x > 0) ? this._mx : (-this._mx);
					this._my = (this.y < h && this.y > 0) ? this._my : (-this._my);
					this.x += this._mx / 2;
					this.y += this._my / 2;
				}


				this.drawLine = drawLine;
				function drawLine(ctx, _circle) {
					var dx = this.x - _circle.x;
					var dy = this.y - _circle.y;
					var d = Math.sqrt(dx * dx + dy * dy);
					if (d < 150) {
						ctx.beginPath();
						ctx.moveTo(this.x, this.y);//start
						ctx.lineTo(_circle.x, _circle.y);//end
						ctx.strokeStyle = 'rgba(204, 204, 204, 0.3)';
						ctx.stroke();
					}
				}
			}

			function currentCircle(x, y) {
				circle.call(this, x, y);

				this.drawCircle = drawCircle;
				function drawCircle(ctx) {
					ctx.beginPath();
					ctx.arc(this.x, this.y, 8, 0, 360);
					ctx.fillStyle = 'rgba(255, 77, 54, 0.6)';
					ctx.fill();
				}
			}
		</script>
		<!-- Navigation -->
<nav class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">
        <div class="navbar-header page-scroll">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand page-scroll" href="#home">DeepEA</a>
        </div>
        <!-- Collect the nav links, forms, and other content for toggling -->
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <li><a class="page-scroll" href="#about">About</a></li>
                <li><a class="page-scroll" href="#installation">Installation</a></li>
                <li><a class="page-scroll" href="#pre-analysis">Pre-analysis</a></li>
                <li><a class="page-scroll" href="#core-analysis">Core-analysis</a></li>
                <li><a class="page-scroll" href="#advanced-analysis">Advanced-analysis</a></li>
                <li><a class="page-scroll" href="#examples">Examples</a></li>
                <li><a class="page-scroll" href="#how-to-access-help">Contact</a></li>
                <li><a class="page-scroll" href="https://github.com/cma2015/DeepEA" target="_blank">Source code</a></li>
            </ul>
        </div>
        <!-- /.navbar-collapse -->
    </div>
    <!-- /.container -->
</nav>

<section id="home">
<!-- Header -->

    <div class="intro-header">

        <div class="container">

            <div class="row">
                <div class="col-lg-12">
                    <div class="intro-message">
                        <h1>Welcome to DeepEA</h1>
                        <h3>deep analysis of epitranscriptome sequencing data for non-specialists</h3>
							<div class="panel">

  <div class="button-effect">
    <a class="effect effect-2" style='text-decoration:none;' data-toggle="modal" href="#login-modal" title="Test Data">Test Data</a>
    <a class="effect effect-2" style='text-decoration:none;' href="https://deepea.nwafu.edu.cn" tilte="DeepEA demo" target="_blank">DeepEA demo</a>
    <a class="effect effect-2" style='text-decoration:none;' href="https://youtu.be/RK-0X6saetM" tilte="Video tutorial" target="_blank">Video tutorial</a>
  </div>
</div>
<div class="modal" id="login-modal">
	<a class="close" data-dismiss="modal">×</a>
	<h1>Download</h1>
	<div class="button-effect">
	<a class="effect effect-2"  style='text-decoration:none;' href="https://deepea.nwafu.edu.cn/static/test_data/1-PRE-ANALYSIS.zip">PRE-ANALYSIS</a></br></br>
	<a class="effect effect-2" style='text-decoration:none;' href="https://deepea.nwafu.edu.cn/static/test_data/2-CORE-ANALYSIS.zip">CORE-ANALYSIS</a></br></br>
	<a class="effect effect-2" style='text-decoration:none;'  href="https://deepea.nwafu.edu.cn/static/test_data/3-ADVANCED-ANALYSIS.zip">ADVANCED-ANALYSIS</a></br></br>
	</div>
</div>

                    </div>

                </div>

            </div>


        </div>
        <!-- /.container -->

    </div>

    <!-- /.intro-header -->
</section>


        <div class="content-section-a">
            <div class="container">
               <div class="lead">


<h1 id="about">About</h1>
<p><img src="assets/img/pulls_pea.svg" alt="docker pulls" />
<img src="assets/img/star_pea.svg" alt="docker stars" /></p>

<p>DeepEA is a Galaxy-based platform for exploring epitranscriptome sequencing. Currently it integrated a suit of tools required for epitranscriptome analysis including Data Preparation, Quality Control, CMR (chemical modifications of RNA) profiling and analysis, Machine learning-based CMR prediction, CMR annotation and visualization and multi-Omics integration analysis. DeepEA provided an easy-to-use interface and detailed description to help researchers without any programming background to analyze epitranscriptome sequencing data easily. DeepEA project is hosted on https://github.com/cma2015/DeepEA. The DeepEA demo server can be accessed via https://deepea.nwafu.edu.cn.</p>
<div class="box">
			<div class="box-1">
				<ul>
					<li>
						<img src="assets/img/Functions_in_DeepEA.jpg" alt="Functions in DeepEA"></img>
						<h2>Functions in DeepEA</h2>
					</li>
					<li>
						<img src="assets/img/DeepEA_logo.jpg" alt="DeepEA logo"></img>
						<h2>DeepEA logo</h2>
					</li>
				</ul>
			</div>
			<div class="box-2">
				<ul>

				</ul>
			</div>
			<div class="box-3">
				<span class="prev"> < </span>
				<span class="next"> > </span>
			</div>
		</div>
<div id="top"></div>


<h1 id="table-of-contents">Table of Contents</h1>
<ul>
  <li><a href="#installation">Installation</a>
    <ul>
        <li><a href="#docker-installation">Docker installation</a></li>
        <li><a href="#deepea-quick-start">DeepEA quick start</a></li>
        <li><a href="#how-to-upload-data">How to upload data</a></li>
        <li><a href="#browser-compatibility">Browser compatibility</a></li>
    </ul>

  </li>
  <li><a href="#pre-analysis">Pre-analysis</a>
    <ul>
      <li><a href="#data-preparation">Data Preparation</a></li>
      <li><a href="#quality-control">Quality control</a></li>
    </ul>
  </li>
  <li><a href="#core-analysis">Core-analysis</a>
    <ul>
      <li><a href="#cmr-profiling">CMR Profiling</a>
        <ul>
          <li><a href="#align-reads-to-genome">Align Reads to Genome</a></li>
          <li><a href="#cmr-calling">CMR Calling</a></li>
        </ul>
      </li>
      <li><a href="#machine-learning-based-cmr-prediction">Machine Learning-based CMR Prediction</a>
        <ul>
          <li><a href="#feature-encoding">Feature encoding</a></li>
          <li><a href="#feature-selection">Feature selection</a></li>
          <li><a href="#machine-learning-based-classification">Machine learning-based classification</a></li>
        </ul>
      </li>
    </ul>
  </li>
  <li><a href="#advanced-analysis">Advanced-analysis</a>
    <ul>
      <li><a href="#cmr-annotation-and-visualization">CMR Annotation and Visualization</a></li>
      <li><a href="#multi-omics-integration-analysis">Multi-omics Integration Analysis</a></li>
    </ul>
  </li>
  <li><a href="#examples">Examples</a></li>
  <li><a href="#how-to-access-help">How to access help</a></li>
</ul>

<h1 id="installation">Installation</h1>
<h2 id="docker-installation">Docker installation</h2>

<blockquote>
  <p>Official installation tutorial is available at <a href="https://docs.docker.com/install" target="_blank">here</a></p>
</blockquote>

<h3 id="for-windows-test-on-windows-10-enterprise-version">For Windows (Test on Windows 10 Enterprise version):</h3>
<ul>
  <li>Download <a href="https://download.docker.com/win/stable/Docker%20for%20Windows%20Installer.exe"  target="_blank">Docker</a> for windows;</li>
  <li>Double click the EXE file to open it;</li>
  <li>Follow the wizard instruction and complete installation;</li>
  <li>Search docker, select <strong><em>Docker for Windows</em></strong> in the search results and clickit.</li>
</ul>

<h3 id="for-mac-os-x-test-on-macos-sierra-version-10126-and-macos-high-sierra-version-10133">For Mac OS X (Test on macOS Sierra version 10.12.6 and macOS High Sierra version 10.13.3):</h3>
<ul>
  <li>Download <a href="https://download.docker.com/mac/stable/Docker.dmg" target="_blank">Docker</a> for Mac OS X</li>
  <li>Double click the DMG file to open it;</li>
  <li>Drag the docker into Applications and complete installation;</li>
  <li>Start docker from Launchpad by click it.</li>
</ul>

<h3 id="for-ubuntu-test-on-ubuntu-1604-lts">For Ubuntu (Test on Ubuntu 16.04 LTS):</h3>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># Update the apt package index:</span>
<span class="nb">sudo </span>apt-get update
<span class="c"># Install packages to allow apt to use a repository over HTTPS:</span>
<span class="nb">sudo </span>apt-get <span class="nb">install</span> <span class="se">\</span>
    apt-transport-https <span class="se">\</span>
    ca-certificates <span class="se">\</span>
    curl <span class="se">\</span>
    software-properties-common
<span class="c"># Add Docker’s official GPG key</span>
curl <span class="nt">-fsSL</span> https://download.docker.com/linux/ubuntu/gpg | <span class="nb">sudo </span>apt-key add -
<span class="nb">sudo </span>apt-key fingerprint 0EBFCD88
<span class="nb">sudo </span>add-apt-repository <span class="se">\</span>
   <span class="s2">"deb [arch=amd64] https://download.docker.com/linux/ubuntu </span><span class="se">\</span><span class="s2">
   </span><span class="si">$(</span>lsb_release <span class="nt">-cs</span><span class="si">)</span><span class="s2"> </span><span class="se">\</span><span class="s2">
   stable"</span>
<span class="nb">sudo </span>apt-get update
<span class="nb">sudo </span>apt-get <span class="nb">install </span>docker-
<span class="c"># Verify that Docker CE is installed correctly by running the hello-world image</span>
<span class="nb">sudo </span>docker run hello-world
</code></pre></div></div>

<h3 id="verify-if-docker-is-installed-correctly">Verify if Docker is installed correctly</h3>

<p>Once Docker installation is completed, we can run <strong>hello-world</strong> image to verify if Docker is installed correctly. Open terminal in Mac OS X and Linux operating system and open CMD for Windows operating system, then type the following command:</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>docker run hello-world
</code></pre></div></div>


<div text-align="center">
    <img src="assets/img/Verify_docker.png"  width="1100px" align="middle"/>
</div>

<p><strong><font color =red>Note</font>: root permission is required for Linux operating system, if you want to manage docker as a non-root user, please refer <a href="https://docs.docker.com/install/linux/linux-postinstall" target="_blank">here</a></strong></p>

<h2 id="deepea-quick-start">DeepEA quick start</h2>

<p>DeepEA has been packaged into a docker image, which can be easily installed as a local server by following command. See following video to see how to install DeepEA locally.</p>

<div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c"># pull DeepEA from docker hub</span>
docker pull malab/deepea
docker run <span class="nt">-it -p 8080:80</span> malab/deepea /bin/zsh
bash /home/DeepEA_v2/galaxy/run.sh
</code></pre></div></div>

<p>Then DeepEA local server can be accessed via <code class="language-plaintext highlighter-rouge">https://localhost:8080</code></p>

<div text-align="center">
    <a href="https://youtu.be/RK-0X6saetM" target="_blank">
        <img src="assets/img/how_to_use_DeepEA.png"  width="1100px" align="middle"/>
    </a>   
</div>



<h2 id="how-to-upload-data">How to upload data</h2>

<p>See following video to see how to upload data into DeepEA server.</p>

<div text-align="center">
    <a href="https://www.youtube.com/watch?v=3WAOd2p1c5w" target="_blank">
        <img src="assets/img/how_to_upload_data.png"  width="1100px" align="middle"/>
    </a>   
</div>

<h2 id="browser-compatibility">Browser compatibility</h2>

<p>DeepEA has been tested in following OS versions and browsers</p>

<div style="text-align:center">
    <table border="2" align="center" cellspacing="0" cellpadding="" id="grid1">
        <tr>
            <td><b>OS</b></td>
            <td><b>Version</b></td>
            <td><b>Chrome</b></td>
            <td><b>Firefox</b></td>
            <td><b>Microsoft Edge</b></td>
            <td><b>Safari</b></td>
        </tr>
        <tr>
            <td>Linux</td>
            <td>Ubuntu 18.04</td>
            <td>&radic;</td>
            <td>&radic;</td>
            <td>n/a</td>
            <td>n/a</td>
        </tr>
        <tr>
            <td>MacOS</td>
            <td>Mojave</td>
            <td>&radic;</td>
            <td>&radic;</td>
            <td>n/a</td>
            <td>&radic;</td>
        </tr>
        <tr>
            <td>Windows</td>
            <td>10</td>
            <td>&radic;</td>
            <td>&radic;</td>
            <td>&radic;</td>
            <td>n/a</td>
        </tr>
    </table>
</div>

<h1 id="pre-analysis">Pre-analysis</h1>

<p><strong>Pre-analysis</strong> module is composed of two sub-modules: <a href="#data-preparation">Data Preparation</a> and <a href="#quality-control">Quality Control</a>. These two sub-modules make up the important components of Epitranscriptome.</p>



<h2 id="data-preparation">Data Preparation</h2>

<p>This sub-module provides three funcitons (see following table for details) to prepare epitranscriptome data.</p>
<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="" id="grid1">
    <tr>
        <td><b>Tools</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>Time (test data)</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Obtain_Genome_or_Transcriptome_Sequences_and_Annotation&version=18.09&__identifer=hu4175p4gw" target="_blank">Obtain Genome Sequences and Annotation</a></td>
        <td>Fetch genome, cDNA, CDS sequences and annotation from Enseml automatically</td>
        <td>Select an Ensembl database; Select a species; Select dada type</td>
        <td>FASTA format sequences or GTF annotation</td>
        <td>Depends on the network speed</td>
        <td>In-house scripts</td>
   </tr>
   <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Obtain_Epitranscriptome_Sequencing_Reads&version=18.09&__identifer=cldn1n7jota" target="_blank">Obtain Epitranscriptome Sequencing and Reads</a></td>
        <td>Fetch epitranscriptome sequencing reads from NCBI's SRA database</td>
        <td>SRR accession or HTTP/FTP link</td>
        <td>Sequencing files in SRA/FASTQ format</td>
        <td>Depends on the network speed</td>
        <td><a href="https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft" target="_blank">SRA Toolkit</a></td>
   </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Convert_Format&version=18.09&__identifer=g9f7k74nmuq" target="_blank">Sequence Data Preprocessing</a></td>
        <td>Convert SRA to FASTQ</td>
        <td>Sequences file in SRA format</td>
        <td>Sequences file in FASTQ format</td>
        <td>~2 mins</td>
        <td><a href="https://www.ncbi.nlm.nih.gov/sra/docs/toolkitsoft" target="_blank">SRA Toolkit</a></td>
   </tr>
</table>
</div>

<h2 id="quality-control">Quality Control</h2>
<p><strong>Quality Control</strong> submodule consisted of a suite of tools focused on different levels of quality assessment including reads quality, alignment quality and CMR quality, Thus, three functions are implemented including <strong>Assess Reads Quality</strong>, <strong>Assess Alignment Quality</strong> and <strong>Assess CMR quality</strong>. The details of as follows:</p>

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="" id="grid2">
    <tr>
        <td><b>Tool</b></td>
        <td><b>Description</b></td>
        <td><b>Main functions</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>Time (test data)</b></td>
        <td><b>Programs</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Assess_Reads_Quality&version=18.09&__identifer=fi48oxvabqm" target="_blank">Assess Reads Quality</a></td>
        <td>This tool integrates FastQC and fastp to perform quality control from raw epitranscriptome sequencing reads</td>
        <td>Reads trimming and generate reads quality report including sequence quality, GC content, etc</td>
        <td>Sequencing reads in FASTQ format</td>
        <td>Clean reads or reads quality report in HTML format</td>
        <td>~40s</td>
        <td><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="_blank">FastQC</a>,<a href="https://github.com/OpenGene/fastp" target="_blank">fastp</a></td>
        <td><a href="https://doi.org/10.1093/bioinformatics/bty560" target="_blank">Chen <em>et al</em>., 2018, Bioinformatics</a>, <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc" target="_blank">Babraham Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Assess_Alignment_Quality&version=18.01&__identifer=kilhidogzl" target="_blank">Assess Alignment Quality</a></td>
        <td>This function is used to generate quality assessment.</td>
        <td>Evaluate alignment quality</td>
        <td>Reads alignment fies in SAM/BAM format</td>
        <td>Alignment quality report in HTML format</td>
        <td>~1 min</td>
        <td><a href="https://github.com/skyhorsetomoon/Trumpet" target="_blank">trumpet</a></td>
        <td><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2266-3" target="_blank">Zhang et al., 2018, BMC Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Assess_CMR_Regions_Quality&version=18.01&__identifer=8b6080rjsk8" target="_blank">Assess CMR Quality</a></td>
        <td>This tool aims to quantify CMR signal strength by counting reads and calculating RPKM in binding site intervals</td>
        <td>Quantify CMRs</td>
        <td>Alignments in SAM/BAM format</td>
        <td>CMR quantification matrix</td>
        <td>~1 min</td>
        <td><a href="https://bioconductor.org/packages/release/bioc/html/DiffBind.html" target="_blank">DiffBind</a></td>
        <td><a href="https://www.frontiersin.org/articles/10.3389/fgene.2015.00169/full" target="_blank">Wu <em>et al</em>., 2016, frontiers in Genetics</a></td>
    </tr>
</table>
</div>

<h1 id="core-analysis">Core-analysis</h1>
<p><strong>Core-analysis</strong> module consists of <strong>CMR Profiling</strong> and <strong>Machine Learning-based CMR Prediction</strong>, of which <strong>CMR Profiling</strong> provides step-by-step functions required for epitranscriptome reads mapping, CMR Calling and differential CMR analysis, while <strong>Machine Learning-based CMR Prediction</strong> is designed to automatically complete the three main steps for constructing a CMR predictor including <strong>Sequence Feature Extraction</strong>, <strong>Feature Engineering</strong> and <strong>Machine Learning-based Classification</strong></p>


<h2 id="align-reads-to-genome">Align Reads to Genome</h2>
<p>Several commonly used aligners are wrapped to align epitranscriptome reads to genome. Currently, <a href="https://ccb.jhu.edu/software/tophat/index.shtml" target="_blank">Tophat2</a>, <a href="https://bowtie-bio.sourceforge.net/bowtie2/index.shtml" target="_blank">Bowtie2</a>, <a href="https://github.com/alexdobin/STAR">STAR</a>, <a href="https://www.ccb.jhu.edu/software/hisat/index.shtml" target="_blank">HISAT2</a> and <a href="https://bio-bwa.sourceforge.net/bwa.shtml" target="_blank">bwa-mem</a> are available.</p>

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="" id="grid3">
    <tr>
        <td><b>Tools</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>Time (test data)</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=tophat2&version=18.09&__identifer=8ug4tmah3wl" target="_blank">Tophat2</a></td>
        <td>Accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions</td>
        <td rowspan="5">Sequencing reads in FASTQ format</td>
        <td rowspan="5">Alignments in SAM/BAM format</td>
        <td>~50s</td>
        <td><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36" target="_blank">Kim <em>et al</em>., 2013, Genome Biology</a></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=bowtie2&version=18.09&__identifer=ngfg3uht6o" target="_blank">Bowtie2</a></td>
        <td>A fast gapped-read alignment tool</td>
        <td>~10 s</td>
        <td><a href="https://www.nature.com/articles/nmeth.1923" target="_blank">Langmead <em>et al</em>., 2012, Nature Methods</a></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=star&version=18.09&__identifer=wekt4r09u9" target="_blank">STAR</a></td>
        <td>An ultrafast universal RNA-seq aligner</td>
        <td>~16s</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/29/1/15/272537" target="_blank">Dobin <em>et al</em>., 2013, Bioinformatics</a></td>
   </tr>
   <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=hisat2&version=18.09&__identifer=ukcle1eozln" target="_blank">HISAT2</a></td>
        <td>A fast spliced aligner with low memory requirements</td>
        <td>~8s</td>
        <td><a href="https://www.nature.com/articles/nmeth.3317" target="_blank">Kim <em>et al</em>., 2015, Nature Methods</a></td>
   </tr>
   <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=bwa-mem&version=18.09&__identifer=t2e00gi0cjp" target="_blank">bwa-mem</a></td>
        <td>Fast and accurate short read alignment with Burrows–Wheeler transform</td>
        <td>~10s</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/25/14/1754/225615" target="_blank">Li <em>et al</em>., 2009, Bioinformatics</a></td>
   </tr>
 </table>
</div>

<h2 id="cmr-calling">CMR Calling</h2>
<p><strong>CMR Calling</strong> implements four pipelines for MeRIP-Seq, m6A miCLIP-Seq, CeU-Seq and RNA-BSSeq, respectively.</p>

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="" id="grid4">
    <tr>
        <td><b>Tools</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>Time (test data)</b></td>
        <td><b>Programs</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Calling_peaks&version=18.09&__identifer=lveislweqml" target="_blank">Peak calling from the MeRIP-Seq data</a></td>
        <td>Identifying enriched genomic regions from MeRIP-Seq experiment</td>
        <td>Alignments of IP and input in SAM/BAM format and Reference genome sequences in FASTA format</td>
        <td>CMR regions in BED format</td>
        <td>~36s</td>
        <td><a href="https://github.com/cma2015/PEA" target="_blank">PEA</a></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/34/21/3747/5021690" target="_blank">Zhai <em>et al</em>., 2018, Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=peakcalling_m5C&version=18.09&__identifer=hsfevs1a0jg" target="_blank">Calling m5C from the RNA-BSseq data</a></td>
        <td>Performing bisulfite sequencing (BS-Seq) read mapping, comprehensive methylation calling using meRanTK</td>
        <td>Sequencing reads in FASTQ format and reference genome sequences in FASTTA format</td>
        <td>m5C sites in BED format</td>
        <td>~10 mins</td>
        <td><a href="https://icbi.at/software/meRanTK">meRanTK</a></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/32/5/782/1744216" target="_blank">Rieder <em>et al</em>., 2016, Bioinformatics</a></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Calling_pseudoU&version=18.09&__identifer=nq46b0yeys" target="_blank">Calling Ψ from CeU-Seq data</a></td>
        <td>Identifying pseudouridylation from CeU-Seq</td>
        <td>Alignments in SAM/BAM format and cDNA sequences in FASTA format</td>
        <td>Pseudoridylation sites in BED format</td>
        <td>~4 mins</td>
        <td>In-house scripts</td>
        <td><a href="https://www.nature.com/articles/nchembio.1836" target="_blank">Li <em>et al</em>., 2015, Nature Chemical Biology</a></td>
    </tr>
</table>
</div>

<h2 id="machine-learning-based-cmr-prediction">Machine Learning-based CMR Prediction</h2>
<p>This sub-module provides a pipeline for transcriptome-wide CMR prediction using machine learning technology. This pipeline is consisted of <a href="#feature-encoding">Feature encoding</a>, <a href="#feature-selection">Feature selection</a> and <a href="#machine-learning-based-classification">Machine learning-based classification</a></p>

<h3 id="feature-encoding">Feature encoding</h3>

<p><strong>Sequence-derived features</strong> (~ 3mins for test data)</p>

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="" id="grid5">
    <tr>
        <td><b>Type</b></td>
        <td><b>Feature</b></td>
        <td><b>Description</b></td>
        <td><b>Dimension</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td rowspan="4">Nucleic acid composition related features</td>
        <td>1mer</td>
        <td>The frequency of mono-nucleotide</td>
        <td>4</td>
        <td rowspan="4"><a href="https://academic.oup.com/bib/article-abstract/20/4/1280/4763667?redirectedFrom=fulltext" target="_blank">Liu <em>et al</em>., 2017, Briefings in Bioinformatics</a></td>
    </tr>
    <tr>
        <td>2mer</td>
        <td>The frequency of di-nucleotide</td>
        <td>16</td>
    </tr>
    <tr>
        <td>3mer</td>
        <td>The frequency of tri-nucleotide</td>
        <td>64</td>
    </tr>
    <tr>
        <td>4mer</td>
        <td>The frequency of tetra-nucleotide</td>
        <td>256</td>
    </tr>
    <tr>
        <td rowspan="6">Autocorrelation-base features</td>
        <td>DAC (<b><u>D</u></b>inucleotide-based <b><u>a</u></b>uto <b><u>c</u></b>ovariance)</td>
        <td>DAC measures the correlation of the same physicochemical index between two dinucleotides seperated by a distance of <i>d</i> along the sequence</td>
        <td><i>N</i>*<i>d</i></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/25/20/2655/194053" target="_blank">Dong <em>et al</em>., 2009, Bioinformatics</a> <a href="https://academic.oup.com/nar/article/36/9/3025/1104168" target="_blank">Guo <em>et al</em>., 2008, Nucleic Acids Research</a> <a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158" target="_blank">Chen <em>et al</em>., 2015, Bioinformatics</a></td>
    </tr>
    <tr>
        <td>DCC (<b><u>D</u></b>inucleotide-based <b><u>c</u></b>ross <b><u>c</u></b>ovariance)</td>
        <td>DCC measures the correlation of two different physicochemical indices between two dinucleotides separated by a distance of <i>d</i> nucleotide acids along the sequence</td>
        <td><i>N</i>*(<i>N-1</i>)*<i>d</i></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/25/20/2655/194053" target="_blank">Dong <em>et al</em>., 2009, Bioinformatics</a> <a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158" target="_blank">Chen <em>et al</em>., 2015, Bioinformatics</a></td>
    </tr>
    <tr>
        <td>DACC (<b><u>D</u></b>inucleotide-based <b><u>a</u></b>uto-<b><u>c</u></b>ross <b><u>c</u></b>ovariance)</td>
        <td>A combination of DAC and DCC</td>
        <td><i>N</i>*<i>N</i>*<i>d</i></td>
         <td><a href="https://academic.oup.com/bioinformatics/article/25/20/2655/194053" target="_blank">Dong <em>et al</em>., 2009, Bioinformatics</a> <a href="https://academic.oup.com/nar/article/36/9/3025/1104168" target="_blank">Guo <em>et al</em>., 2008, Nucleic Acids Research</a> <a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158" target="_blank">Chen <em>et al</em>., 2015, Bioinformatics</a></td>
    </tr>
    <tr>
        <td>MAC (<b><u>M</u></b>oran <b><u>a</u></b>uto<b><u>c</u></b>orrelation)</td>
        <td>MAC measures the correlation of the same properties between two residues separated by a distance of <i>d</i> along the sequence</td>
        <td><i>N</i>*<i>k</i>*<i>d</i></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158" target="_blank">Chen <em>et al</em>., 2015, Bioinformatics</a> <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/bip.360270308" target="_blank">Horne <em>et al</em>., 1988, Biopolymers</a></td>
    </tr>
    <tr>
        <td>GAC (<b><u>G</u></b>eary <b><u>a</u></b>uto<b><u>c</u></b>orrelation)</td>
        <td>GAC measures the correlation of the same properties between two residues separated by a distance of <i>d</i> along the sequence</td>
        <td><i>N</i>*<i>k</i>*<i>d</i></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158" target="_blank">Chen <em>et al</em>., 2015, Bioinformatics</a> <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ajpa.20250" target="_blank">Sokal <em>et al</em>., 2006, American
journal of physical anthropology</a></td>
    </tr>
    <tr>
        <td>NMBAC (<b><u>N</u></b>ormalized <b><u>M</u></b>oreau-<b><u>B</u></b>roto <b><u>a</u></b>uto<b><u>c</u></b>orrelation)</td>
        <td>It measures the corelatio of the same properties between two residues separated by a distance of <i>lag</i> along the sequence.</td>
        <td><i>N</i>*<i>d</i></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158" target="_blank">Chen <em>et al</em>., 2015, Bioinformatics</a> <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ajpa.20250" target="_blank">Feng <em>et al</em>., 2000, Journal of protein chemistry</a></td>
    </tr>
    <tr>
        <td rowspan="2">Pseudo nucleotide composition</td>
        <td>PC-PseDNC-General</td>
        <td>General parallel correlatio pseudo dinucleotide composition</td>
        <td>16+&lambda;</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158" target="_blank">Chen <em>et al</em>., 2015, Bioinformatics</a></td>
    </tr>
    <tr>
        <td>SC-PseDNC-General</td>
        <td>General series correlation pseudo dinucleotide composition</td>
        <td>16+&lambda;</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/31/1/119/2366158" target="_blank">Chen <em>et al</em>., 2015, Bioinformatics</a></td>
    </tr>
    <tr>
        <td rowspan="2">Binary encoding</td>
        <td>Binary encoding</td>
        <td>A, C, G, U are encoded with (1, 0, 0, 0), (0, 1, 0, 0), (0, 0, 1, 0), (0, 0, 0, 1), respectively</td>
        <td>4*<i>L</i></td>
        <td><a href="https://academic.oup.com/bioinformatics/article/34/21/3747/5021690" target="_blank">Zhai <em>et al</em>., 2018, Bioinformatics</a> <a href="https://www.frontiersin.org/articles/10.3389/fpls.2018.00519/full" target="_blank">Song <em>et al</em>., 2018, Frontiers in Plant Science</a></td>
    </tr>
    <tr>
        <td>Binary encoding based on structural chemical properties</td>
        <td>A, C, G, U are encoded as a vector of three features (1,1,1), (0,1,0), (1,0,0) and (0,0,1), respectively</td>
        <td>3*<i>L</i></td>
        <td><a href="https://academic.oup.com/nar/article/47/7/e41/5319125#133859200" target="_blank">Chen <em>et al</em>., 2019, Nucleic Acids Research</a></td>
    </tr>
</table>
</div>

<p><strong>Note:</strong></p>
<ul>
  <li><em>N</em> is the number of physicochemical indices, here is 22;</li>
  <li><em>L</em> is the sequence length;</li>
  <li><em>d</em> is the length between two seperated nucleotides.</li>
  <li><i>λ</i> is an interger, representing the highest counted rank of the correlation along a RNA sequence.</li>
</ul>

<p><strong>Genomic-derived features</strong> (~ 3mins for test data)</p>
<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="" id="grid6">
     <tr>
        <td><b>Type</b></td>
        <td><b>Feature</b></td>
        <td><b>Description</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td rowspan="12">Dummy variables indicating whether the site is overlapped to the topological on the major RNA transcript</td>
        <td>Five prime UTR</td>
        <td>Binary value indicating whether CMR is overlapped with five prime UTR</td>
        <td rowspan="25"><a href="https://academic.oup.com/nar/article/47/7/e41/5319125#133859200" target="_blank">Chen <em>et al</em>., 2019, Nucleic Acids Research</a></td>
    </tr>
    <tr>
        <td>Three prime UTR</td>
        <td>Binary value indicating whether CMR is overlapped with three prime UTR</td>
    </tr>
    <tr>
        <td>Stop codons</td>
        <td>Binary value indicating whether CMR is overlapped with stop codons</td>
    </tr>
    <tr>
        <td>Start codons</td>
        <td>Binary value indicating whether CMR is overlapped with start codons</td>
    </tr>
    <tr>
        <td>Downstream 100bp of TSS (Transcription Start Sites)</td>
        <td>Binary value indicating whether CMR is located at the downstream 100bp of TSS</td>
    </tr>
    <tr>
        <td>Exons containing stop codons</td>
        <td>Binary value indicating whether stop codons are located at the exon containing CMR</td>
    </tr>
    <tr>
        <td>Alternative exons</td>
        <td>Binary value indicating the exon containing CMR is an alternative exon</td>
    </tr>
    <tr>
        <td>Constitutive exons</td>
        <td>Binary value indicating the exon containing CMR is a constitutive exon</td>
    </tr>
    <tr>
        <td>Internal exons</td>
        <td>Binary value indicating the exon containing CMR is an internal exon</td>
    </tr>
    <tr>
        <td>Long exons</td>
        <td>Binary value indicating the exon containing CMR is a long exon (exon length &gt;= 400bp)</td>
    </tr>
    <tr>
        <td>Five prime 500bp of the last exon</td>
        <td>Binary value indicating whether the exon containing CMR is located at the five prime 500bp of the last exon</td>
    </tr>
    <tr>
        <td>Five prime 500bp of the last exon containing stop codons</td>
        <td>Binary value indicating whether the exon containing CMR is located at the five prime 500bp of the last exon containing stop codons</td>
    </tr>
    <tr>
        <td rowspan="3">Relative position on the region</td>
        <td>Relative position on five prime UTR</td>
        <td>An integer value indicating the relative position of CMR on five prime UTR</td>
    </tr>
    <tr>
        <td>Relative position on three prime UTR</td>
        <td>An integer value indicating the relative position of CMR on three prime UTR</td>
    </tr>
    <tr>
        <td>Relative position on exon</td>
        <td>An integer value indicating the relative position of CMR on the exon containing CMR</td>
    </tr>
    <tr>
        <td rowspan="3">The region length in bp</td>
        <td>The length of five prime UTR</td>
        <td>An interger value indicating the length of the five prime UTR containing CMR</td>
    </tr>
    <tr>
        <td>The length of three prime UTR</td>
        <td>An interger value indicating the length of three three prime UTR containing CMR</td>
    </tr>
    <tr>
        <td>Mature transcript length</td>
        <td>An interger value indicating the length of the mature transcript containing CMR</td>
    </tr>
    <tr>
        <td rowspan="3">Nucleotide distances toward the splicing junctions or the nearest neighboring sites</td>
        <td>Distance to the five prime splicing junction</td>
        <td>An interger indicating the distance to the five prime splicing junction</td>
    </tr>
    <tr>
        <td>Distance to the three prime splicing junction</td>
        <td>An interger indicating the distance to the three prime splicing junction</td>
    </tr>
    <tr>
        <td>Distance to the closest neighbor truncated at 200bp</td>
        <td>An interger indicating the distance to the closest neighbor truncated at 200bp</td>
    </tr>
    <tr>
        <td rowspan="4">Attributes of the genes of transcripts</td>
        <td>sncRNA</td>
        <td>Binary value indicating if the gene containing CMR is a sncRNA</td>
    </tr>
    <tr>
        <td>lncRNA</td>
        <td>Binary value indicating if the gene containing CMR is a lncRNA</td>
    </tr>
    <tr>
        <td>House keeping genes</td>
        <td>Binary value indicating if the gene containing CMR is a house keeping gene</td>
    </tr>
    <tr>
        <td>miRNA targeted genes</td>
        <td>Binary value indicating if the gene containing CMR is an miRNA target gene</td>
    </tr>
</table>
</div>

<h3 id="feature-selection">Feature selection</h3>

<p>The function aims to perform feature selection, currently, seven commonly used algorithms implemented in R package <a href="https://cran.r-project.org/web/packages/caret/vignettes/caret.html">"caret"</a> are wrapped.</p>

<div style="text-align:center">
    <table border="2" align="center" cellspacing="0" cellpadding="" id="grid6">
         <tr>
            <td><b>Algorithms</b></td>
            <td><b>Description</b></td>
            <td><b>Parameters</b></td>
            <td><b>Time (test data)</b></td>
        </tr>
        <tr>
            <td>Remove redundant features</td>
            <td>Removing redundant features according to their perason correlatio coefficient</td>
            <td>Pearson correlation coefficient threshold<br></td>
            <td>~10s</td>
        </tr>
        <tr>
            <td>Variable importance from machine learning algorithms</td>
            <td>Evaluate the contribution of each feature to the machine learning algorithms</td>
            <td>The number of selected features; Select a machine learning algorithm</td>
            <td>~2 mins</td>
        </tr>
        <tr>
            <td>Recursive feature elimination (automatic feature selection)</td>
            <td>This algorithm performs backwards selection of predictors based on predictor importance ranking. The predictors are ranked and the less important ones are sequentially eliminated prior to modeling. The goal is to find a subset of predictors that can be used to produce an accurate model.</td>
            <td>The number of most important features the rfe should iterate;<br>The external resampling method</td>
            <td>~3 mins</td>
        </tr>
        <tr>
            <td>Least Absolute Shrinkage and Selection Operator (LASSO) regression</td>
            <td>Using LASSO cross-validation to evaluate feature importance</td>
            <td>Loss to use for cross-validation;<br>Standardize;</br>Parallel</br></td>
            <td>~1 mins</td>
        </tr>
        <tr>
            <td>Genetic algorithm</td>
            <td>Supervised feature selection using genetic algorithms</td>
            <td>The external resampling method;<br>The number of interations;<br>The number of repeats;<br></td>
            <td>~1 mins</td>
        </tr>
        <tr>
            <td>Simulated annealing</td>
            <td>This algorithm conducts a supervised binary search of the predictor space using simulated annealing (SA)</td>
            <td>The external resampling method<br>The number of interations</br>The number of repeats<br></td>
            <td>~2 mins</td>
        </tr>
        <tr>
            <td>Feature selection based on dropout loss</td>
            <td>Based on random forest algorithm, this algorithm estimates the variable dropout loss for each feature</td>
            <td>The number of trees</td>
            <td>~3 mins</td>
        </tr>
        
    </table>
</div>

<h3 id="machine-learning-based-classification">Machine learning-based classification</h3>
<p>In this module, several commonly-used machine learning classification algorithms are implemented to construct a CMR predictor. The details are described as follows:</p>

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="" id="grid7">
        <tr width="100%">
            <td width="12%"><b>Algorithms</b></td>
            <td width="30%"><b>Description</b></td>
            <td colspan="4"><b>Parameters</b></td>
        </tr>
        <tr>
        <td>Random Forest (RF)</td>
        <td>RF is an ensemble learning method for classification, regression and other tasks that operates by constructing a multitude of decision trees at training time and outputting the class that is the mode of the classes (classification) or mean prediction (regression) of the individual trees</td>
        <td colspan="4"> Number of trees to grow;<br />  The number of threads used for parallel computation.</td>
    </tr>
    <tr>
        <td>Support Vector Machine (SVM)</td>
        <td>Given a set of training examples, each marked as belonging to one or the other of two categories, an SVM training algorithm builds a model that assigns new examples to one category or the other, making it a non-probabilistic binary linear classifier (although methods such as Platt scaling exist to use SVM in a probabilistic classification setting)</td>
        <td>Polynomial kernel:<br /> degree;<br />  gamma;<br /> coef0</td>
        <td>Radial basis kernel:<br /> gamma<br /></td>
        <td>Sigmoid kernel:<br /> gamma<br /></td>
        <td>Linear kernel:<br /> / </td>
    </tr>
    <tr>
        <td>Decision Tree</td>
        <td>A decision tree is a decision support tool that uses a tree-like model of decisions and their possible consequences, including chance event outcomes, resource costs, and utility. It is one way to display an algorithm that only contains conditional control statements</td>
        <td colspan="4"> Number of trees to grow;<br />  The number of threads used for parallel computation.</td>
    </tr>
    <tr>
        <td>XGBoost</td>
        <td>It provides a gradient boosting framework for classification</td>
        <td colspan="4"> Learning rate<br />  Number of trees to grow<br />  The number of threads used for parallel computation<br /></td>
    </tr>
    <tr>
        <td>Logistic Regression</td>
        <td>In statistics, the logistic model (or logit model) is used to model the probability of a certain class or event existing such as pass/fail, win/lose, alive/dead or healthy/sick</td>
        <td colspan="4"> Optimization algorithm<br />  The number of threads used for parallel computation<br /></td>
    </tr>
</table>
</div>

<h1 id="advanced-analysis">Advanced-analysis</h1>
<p>This module provides comprehensive CMR annotation and interactive visualization, both HTML format with interactive figures and tables and PDF format with publication-ready figures are available to analyze the distribution of CMRs in the genome and transcriptome.</p>

<h2 id="cmr-annotation-and-visualization">CMR Annotation and Visualization</h2>

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="" id="grid8">
    <tr>
        <td><b>Functions</b></td>
        <td><b>Description</b></td>
        <td><b>Input</b></td>
        <td><b>Output</b></td>
        <td><b>Reference</b></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=CMR_distribution&version=18.09&__identifer=k8f29ft0p3m" target="_blank">CMRs Distribution</a></td>
        <td>The distribution of CMRs in the genome and transcriptome including the number of peaks in genomic features, the regions of enrichment of CMRs within transcripts, the enrichment of CMRs in transcriptional start/stop site and the enrichment of CMRs in splicing sites;</td>
        <td> CMR regions in BED format<br />   Genome annotation in GTF/GFF format</td>
        <td>Comprehensive overview of CMRs distribution in HTML or PDF format</td>
        <td>In-house scripts</td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=De_novo_motif_discovery&version=18.09&__identifer=665urffrkm7" target="_blank">De-novo Motif Discovery</a></td>
        <td>This function integrates MEME-ChIP, DREME and HOMER to perform <i>de-novo</i> motif discovery</td>
        <td>Positive sequences in FASTA format</td>
        <td>The discovered motifs in HTML format</td>
        <td><a href="https://academic.oup.com/bioinformatics/article/27/12/1653/257754" target="_blank">Timothy <i>et al</i>., 2011, Bioinformatics</a><br /><a href="https://academic.oup.com/bioinformatics/article/27/12/1696/255896" target="_blank">Philip <i>et al</i>., 2011, Bioinformatics</a><br /><a href="https://doi.org/10.1016/j.molcel.2010.05.004" target="_blank">Heinz <i>et al</i>., 2010, Molecular Cell</a></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Motif_enrichment&version=18.09&__identifer=vnh22wb45xa" target="_blank">Motif Enrichment</a></td>
        <td>This function integrates AME and CentriMo to perform motif enrichment</td>
        <td> Sequence file in FASTA format<br /> Motif file in MEME format</td>
        <td>The motif enrichment in HTML format</td>
        <td><a href="https://www.biomedcentral.com/1471-2105/11/165" target="_blank">Robert <i>et al</i>., 2010, BMC Bioinformatics</a><br /> <a href="https://academic.oup.com/nar/article/40/17/e128/2411117" target="_blank">Timothy <i>et al</i>., 2012, Nucleic Acids Research</a><br /></td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Sequence_logo_visualization&version=18.09&__identifer=ao9kcke4bqw" target="_blank">Sequence Logo Visualization</a></td>
        <td>Visualize motifs in sequences</td>
        <td>Sequences in FASTA format</td>
        <td>Motif logo in PDF format</td>
        <td>In-house scripts</td>
    </tr>
    <tr>
        <td><a href="https://deepea.nwafu.edu.cn/?tool_id=Functional_enrichment_analysis&version=18.09&__identifer=xk2zaqgm7ga" target="_blank">Functional Enrichment Analysis</a></td>
        <td>Perform GO or KEGG enrichment analysis for any species</td>
        <td>Gene list</td>
        <td>The enriched GO/KEGG terms</td>
        <td><a href="https://doi.org/10.1089/omi.2011.0118" target="_blank">Yu <i>et al</i>., 2012, OMICS</a></td>
    </tr>
</table>
</div>

<h2 id="multi-omics-integration-analysis">Multi-omics Integration Analysis</h2>
<p>This functions is designed to correlate CMR-realted gene with different type of genomic features including <strong>Gene length analysis</strong>, <strong>Exon length analysis</strong>, <strong>Intron length analysis</strong>, <strong>Exon number analysis</strong> and <strong>GC content analysis</strong>. To be specific, we split the  genome in a sliding window of 100 (default) adjacent genes with step size of 10 (default), and calculate the frequency of CMR genes in each window (100 adjacent genes)</p>

<div style="text-align:center">
<table border="2" align="center" cellspacing="0" cellpadding="" id="grid9">
    <tr>
        <td><b>Genomic features</b></td>
        <td><b>Correlation</b></td>
        <td><b>Difference</b></td>
    </tr>
    <tr>
        <td>Gene length analysis</td>
        <td><img align="middle" src="assets/img/Correlation_CMR_gene_length.png" alt="Correlation of CMR-related genes with gene length analysis" height="300px" width="300px" valign="middle" /></td>
        <td><img align="middle" src="assets/img/Difference_CMR_gene_length.png" alt="Difference of CMR-related genes with gene length analysis" height="200px" width="200px" valign="middle" /></td>
    </tr>
    <tr>
        <td>Exon length analysis</td>
        <td><img align="middle" src="assets/img/Correlation_CMR_exon_length.png" alt="Correlation of CMR-related genes with exon length analysis" height="300px" width="300px" valign="middle" /></td>
        <td><img align="middle" src="assets/img/Difference_CMR_exon_length.png" alt="Difference of CMR-related genes with exon length analysis" height="200px" width="200px" valign="middle" /></td>
    </tr>
    <tr>
        <td>Intron length analysis</td>
        <td><img align="middle" src="assets/img/Correlation_CMR_intron_length.png" alt="Correlation of CMR-related genes with intron length analysis" height="300px" width="300px" valign="middle" /></td>
        <td><img align="middle" src="assets/img/Difference_CMR_intron_length.png" alt="Difference of CMR-related genes with intron length analysis" height="200px" width="200px" valign="middle" /></td>
    </tr>
    <tr>
        <td>Exon number analysis</td>
        <td><img align="middle" src="assets/img/Correlation_CMR_exon_number.png" alt="Correlation of CMR-related genes with exon number analysis" height="300px" width="300px" valign="middle" /></td>
        <td><img align="middle" src="assets/img/Difference_CMR_exon_number.png" alt="Difference of CMR-related genes with exon number analysis" height="200px" width="200px" valign="middle" /></td>
    </tr>
    <tr>
        <td>GC content analysis</td>
        <td><img align="middle" src="assets/img/Correlation_CMR_GC_content.png" alt="Correlation of CMR-related genes with GC content analysis" height="300px" width="300px" valign="middle" /></td>
        <td><img align="middle" src="assets/img/Difference_CMR_GC_content.png" alt="Difference of CMR-related genes with GC content analysis" height="200px" width="200px" valign="middle" /></td>
    </tr>
</table>
</div>

<h1 id="examples">Examples</h1>
<ul>
  <li><a href="./demo_output/Assess_Alignment_Quality.html" target="_blank">Assess Alignment Quality</a></li>
  <li><a href="./demo_output/CMR_distribution_single_case.html" target="_blank">CMRs distribution</a></li>
  <li><a href="./demo_output/dreme.html" target="_blank">De-novo motif discovery for dreme</a></li>
  <li><a href="./demo_output/meme-chip.html" target="_blank">De-novo motif discovery for meme-chip</a></li>
  <li><a href="./demo_output/CMR_genomic_features_for_gene_length.html" target="_blank">Correlate CMR with gene length</a></li>
  <li><a href="./demo_output/CMR_genomic_features_for_exon_length.html" target="_blank">Correlate CMR with exon length</a></li>
  <li><a href="./demo_output/CMR_genomic_features_for_intron_length.html" target="_blank">Correlate CMR with intron length</a></li>
  <li><a href="./demo_output/CMR_genomic_features_for_exon_number.html" target="_blank">Correlate CMR with exon number</a></li>
  <li><a href="./demo_output/CMR_genomic_features_for_GC_content.html" target="_blank">Correlate CMR with GC content</a></li>
</ul>

<h1 id="how-to-access-help">How to access help</h1>
<ul>
  <li>If users encounter any bugs or issues, feel free to leave a message at Github <a href="https://github.com/cma2015/DeepEA/issues" target="_blank">issues</a>. We will try our best to deal with all issues as soon as possible.</li>
  <li>In addition, if any suggestions are available, feel free to contact: Jingjing Zhai <a href="mailto:zhaijingjing603@gmail.com" target="_blank">zhaijingjing603@gmail.com</a> or Chuang Ma <a href="mailto:chuangma2006@gmail.com" target="_blank">chuangma2006@gmail.com</a></li>
</ul>

                </div>
            </div>
</div>

        <!-- Footer -->
<footer>
    <div class="container">
        <div class="row">
            <div class="col-md-8">
                <ul class="list-inline">
                    <li><a href="#home">Home</a></li>
                    <li class="footer-menu-divider">&sdot;</li>
                    <li><a href="#about">About</a></li>
                    <li class="footer-menu-divider">&sdot;</li>
                    <li><a href="#pre-analysis">Pre-analysis</a></li>
                    <li class="footer-menu-divider">&sdot;</li>
                    <li><a href="#core-analysis">Core-analysis</a></li>
                    <li class="footer-menu-divider">&sdot;</li>
                    <li><a href="#advanced-analysis">Advanced-analysis</a></li>
                    <li class="footer-menu-divider">&sdot;</li>
                    <li><a href="#examples">Examples</a></li>
                    <li class="footer-menu-divider">&sdot;</li>
                    <li><a href="https://github.com/cma2015/DeepEA">GitHub</a></li>
                </ul>
            </div>
        </div>
    </div>
</footer>

        <!-- jQuery Version 1.11.0 -->
<script src="assets/js/jquery-1.11.0.js"></script>

<!-- Bootstrap Core JavaScript -->
<script src="assets/js/bootstrap.min.js"></script>
<script type="text/javascript">
grid("grid1","#fff","#F5F5F5","#FFFFCC","#0066CC");
grid("grid2","#fff","#F5F5F5","#FFFFCC","#0066CC");
grid("grid3","#fff","#F5F5F5","#FFFFCC","#0066CC");
grid("grid4","#fff","#F5F5F5","#FFFFCC","#0066CC");
grid("grid5","#fff","#F5F5F5","#FFFFCC","#0066CC");
grid("grid6","#fff","#F5F5F5","#FFFFCC","#0066CC");
grid("grid7","#fff","#F5F5F5","#FFFFCC","#0066CC");
grid("grid8","#fff","#F5F5F5","#FFFFCC","#0066CC");
grid("grid9","#fff","#F5F5F5","#FFFFCC","#0066CC");
</script>
<!-- Custom Theme JavaScript -->
<script src="assets/js/landing-page.js"></script>

	</body>
</html>
