## DeepEA

- DeepEA is a Galaxy-based framework for exploration and large-scale analysis of epitranscriptome sequencing data. **DeepEA** is an upgrated version of [PEA](https://https://github.com/cma2015/PEA), which is a special designed R toolkit for plant epitranscriptome analysis. In the current version of DeepEA, a variety of modules and functions including **data preparation**, **quality control**, **CMR (chemical modifications of RNA) profiling and analysis**, **CMR annotation**, **CMR prediction** and related **visualization** are implemented to enable users perform large-scale, reproducible epitranscriptome analysis. In addition, by taking advantages of multiple instance learning (MIL) algorithms, DeepEA can precisely localize CMRs in peak regions. This can reduce the false positive CMRs generated by MeRIP-Seq/m6A-Seq/m5C-RIP-Seq, as which can localize CMRs to transcript regions of 100-200 nucleotide long but fail to accurately identify CMRs at single-nucleotide solution. The deePEA project is hosted on https://github.com/cma2015/DeepEA. The DeepEA web server can accessed via http://bioinfo.nwafu.edu.cn:4003. To enable uses build a local DeepEA server, we also packaged DeepEA into a [Docker image](https://hub.docker.com/r/malab/deepea).

## How to use DeepEA

- More details about how to use deepEA are presented at https://github.com/cma2015/DeepEA/blob/master/tutorial/content/DeepEA-user-manual.rst
- Test datasets can de downloaded from http://bioinfo.nwafu.edu.cn:4003/library/list#folders/F5969b1f7201f12ae

## How to access help
* If users encounter any bugs or issues, feel free to leave a message at Github [issues](<https://github.com/cma2015/deepEA/issues>). We will try our best to deal with all issues as soon as possible.
* In addition, if any suggestions are available, feel free to contact: __Jie Song__ <q2516581@126.com> or __Jingjing Zhai__ <zhaijingjing603@gmail.com> 


