---
title: Genomic Feature Analysis of CMR
output: 
  flexdashboard::flex_dashboard:
    orientation: rows
    social: menu
    source_code: embed
    theme: cosmo
---

```{r setup, include=FALSE}
library(flexdashboard)
library(rtracklayer)
library(plotly)
library(ggplot2)
library(seqinr)
library(GenomicFeatures)
library(knitr)
library(kableExtra)
options(scipen = 20)
knitr::opts_chunk$set(echo = TRUE,
                      warning = FALSE,
                      message = FALSE)
```


```{r echo = FALSE, warning = FALSE, message = FALSE}
CMRGene <- read.table(file = CMRGeneDir, sep = "\t", header = F, quote = '',
                      stringsAsFactors = F)

GTF <- import(con = GTFDir)
GTF <- subset(GTF, GTF$gene_biotype == "protein_coding")
txdb <- makeTxDbFromGFF(file = GTFDir)

seqNames <- na.omit(as.numeric(as.character(unique(seqnames(GTF)@values))))
GTFdf <- data.frame(GTF)

type1Gene <- subset(CMRGene, CMRGene[,2] >= threshold)$V1
type2Gene <- subset(CMRGene, CMRGene[,2] < threshold)$V1
```


### Introduction
- The function provides the correlation analysis of CMR genes with multiple gene features including **Mean gene length**, **Mean exon number**, **Mean GC content** and **Mean distance to adjacent gene**. To be specific, we split the maize genome in a sliding window of 100 adjacent genes with step size of 10, and calculate the frequency of CMR genes in each window (100 adjacent genes).


Row {data-width=100}
---------------------------------------------------------------

```{r echo = FALSE, warning = FALSE, message = FALSE}
type1.gtf <- GTFdf[which(!is.na(match(GTFdf$gene_id, type1Gene))), ]
type1.exonNumber <- as.numeric(unlist(by(data = type1.gtf,
                                         INDICES = type1.gtf[,"gene_id"],
                                         FUN = function(x) length(na.omit(unique(x[,"exon_number"]))))))

type2.gtf <- GTFdf[which(!is.na(match(GTFdf$gene_id, type2Gene))), ]
type2.exonNumber <- as.numeric(unlist(by(data = type2.gtf,
                                         INDICES = type2.gtf[,"gene_id"],
                                         FUN = function(x) length(na.omit(unique(x[,"exon_number"]))))))

type1Stat <- boxplot.stats(type1.exonNumber)$stats
tmpType1 <- type1.exonNumber[which(type1.exonNumber <= type1Stat[5])]

type2Stat <- boxplot.stats(type2.exonNumber)$stats
tmpType2 <- type2.exonNumber[which(type2.exonNumber <= type2Stat[5])]
```


### The statistics of exon number difference between type 1 and type 2 genes
```{r echo = FALSE, warning=FALSE, message=FALSE}
res <- data.frame(Stats = c("Min", "First quantile", "Median", "Third quantile", "Max"),
                  type1 = type1Stat, type2 = type2Stat)
knitr::kable(x = res, align = 'c') %>%  
  kableExtra::kable_styling(position = 'center', full_width = FALSE, 
                            stripe_color = 'black', latex_options = "bordered")
```


Row {data-width=100}
---------------------------------------------------------------

### Exon number difference between type 1 and type 2 genes
```{r echo = FALSE, warning=FALSE, message=FALSE}

df <- data.frame(type = factor(rep(c("Type 1", "Type 2"),
                                   c(length(tmpType1), length(tmpType2)))),
                 value = c(tmpType1, tmpType2))

p <- ggplot(df, aes(x=type, y=value, fill=type)) + 
  geom_violin(trim = FALSE) +
  theme(axis.title.x = element_blank(),
        #axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        legend.title = element_blank(),
        legend.position = "none")
ggplotly(p)
```


### Exon number density difference between type 1 and type 2 genes
```{r echo = FALSE, warning = FALSE, message = FALSE}
p <- ggplot(df, aes(x=value, fill=type)) + 
  geom_density(alpha = 0.5) +
  theme(axis.title.x = element_blank(),
        #axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        legend.title = element_blank(),
        legend.position = "none")
ggplotly(p)
```
